// @storybook/design-system/dist/components/header/Header.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Header = Header;

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _taggedTemplateLiteral2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteral"));

var _react = _interopRequireDefault(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _styles = require("../shared/styles");

var _WithTooltip = _interopRequireDefault(require("../tooltip/WithTooltip"));

var _Link = require("../Link");

var _Icon = require("../Icon");

var _Nav = require("./Nav");

var _NavItem = require("./NavItem");

var _HeaderContext = require("./HeaderContext");

function _templateObject4() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  width: 3rem;\n  border: none !important;\n  text-decoration: none !important;\n  svg {\n    height: 18px;\n    width: 18px;\n    margin: 0;\n    margin-right: -4px !important;\n  }\n"]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  &:focus svg {\n    color: ", ";\n  }\n"]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  display: none;\n\n  @media (min-width: 375px) {\n    display: inline-flex;\n    align-items: center;\n  }\n"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  transform: scale(0.84);\n  margin-left: -10px;\n  margin-right: -10px;\n\n  ", ";\n  .github-btn {\n    font: bold 14px/14px 'Helvetica Neue', Helvetica, Arial, sans-serif;\n    height: auto;\n    .gh-btn,\n    .gh-count {\n      padding: 4px 8px;\n    }\n  }\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

var GitHubWrapper = _styledComponents["default"].div(_templateObject(), ''
/* Overrides to make a medium sized button */
);

var GithubNavItem = (0, _styledComponents["default"])(_NavItem.NavItem)(_templateObject2());
var MobileMenuTooltip = (0, _styledComponents["default"])(_WithTooltip["default"])(_templateObject3(), _styles.color.secondary);
var Menu = (0, _styledComponents["default"])(_Link.Link)(_templateObject4());

function Header(_ref) {
  var logo = _ref.logo,
      links = _ref.links,
      github = _ref.github,
      mobileMenu = _ref.mobileMenu,
      _ref$inverse = _ref.inverse,
      inverse = _ref$inverse === void 0 ? _HeaderContext.defaultHeaderContext.inverse : _ref$inverse,
      _ref$navBreakpoint = _ref.navBreakpoint,
      navBreakpoint = _ref$navBreakpoint === void 0 ? _HeaderContext.defaultHeaderContext.navBreakpoint : _ref$navBreakpoint,
      props = (0, _objectWithoutProperties2["default"])(_ref, ["logo", "links", "github", "mobileMenu", "inverse", "navBreakpoint"]);
  return /*#__PURE__*/_react["default"].createElement(_HeaderContext.HeaderContext.Provider, {
    value: {
      navBreakpoint: navBreakpoint,
      inverse: inverse
    }
  }, /*#__PURE__*/_react["default"].createElement(_Nav.NavWrapper, props, /*#__PURE__*/_react["default"].createElement(_Nav.Nav, null, /*#__PURE__*/_react["default"].createElement(_Nav.NavGroup, {
    navBreakpoint: navBreakpoint
  }, /*#__PURE__*/_react["default"].createElement(_NavItem.NavItem, null, logo)), /*#__PURE__*/_react["default"].createElement(_Nav.NavGroup, {
    navBreakpoint: navBreakpoint,
    withRightAlignment: true
  }, links, github && /*#__PURE__*/_react["default"].createElement(GithubNavItem, {
    showDesktop: true,
    className: "chromatic-ignore"
  }, /*#__PURE__*/_react["default"].createElement(GitHubWrapper, null, github)), /*#__PURE__*/_react["default"].createElement(_NavItem.NavItem, {
    showMobile: true
  }, /*#__PURE__*/_react["default"].createElement(MobileMenuTooltip, {
    tagName: "span",
    placement: "top",
    trigger: "click",
    tooltip: mobileMenu
  }, /*#__PURE__*/_react["default"].createElement(Menu, {
    secondary: true,
    inverse: inverse,
    icon: 1,
    isButton: true
  }, /*#__PURE__*/_react["default"].createElement(_Icon.Icon, {
    icon: "menu",
    "aria-label": "Menu"
  }))))))));
}

Header.defaultProps = {
  inverse: false
};
