// @storybook/design-system/dist/components/tooltip/Tooltip.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Tooltip = Tooltip;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _taggedTemplateLiteral2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteral"));

var _react = _interopRequireDefault(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireWildcard(require("styled-components"));

var _styles = require("../shared/styles");

function _templateObject4() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n      margin-bottom: ", "px;\n      margin-top: ", "px;\n      margin-left: ", "px;\n      margin-right: ", "px;\n\n      background: rgba(255, 255, 255, 0.97);\n\n      filter: drop-shadow(0px 5px 5px rgba(0, 0, 0, 0.05)) drop-shadow(0 1px 3px rgba(0, 0, 0, 0.1));\n      border-radius: ", "px;\n      font-size: ", "px;\n    "]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n      margin-bottom: ", "px;\n      margin-bottom: ", "px;\n      margin-top: ", "px;\n      margin-top: ", "px;\n      margin-left: ", "px;\n      margin-right: ", "px;\n    "]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  display: ", ";\n  z-index: ", ";\n\n  ", ";\n\n  ", ";\n"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  position: absolute;\n  border-style: solid;\n  /**\n   * Rather than choosing whether or not to render this component in the\n   * render function, the visibility is controlled here. The tooltip library\n   * positions this strangely in some cases if we let the render function\n   * handle this logic.\n   */\n  display: ", ";\n\n  margin-bottom: ", "px;\n  margin-top: ", "px;\n  margin-right: ", "px;\n  margin-left: ", "px;\n\n  bottom: ", "px;\n  top: ", "px;\n  right: ", "px;\n  left: ", "px;\n\n  border-bottom-width: ", "px;\n  border-top-width: ", "px;\n  border-right-width: ", "px;\n  border-left-width: ", "px;\n\n  border-top-color: ", ";\n  border-bottom-color: ", ";\n  border-left-color: ", ";\n  border-right-color: ", ";\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

var ifPlacementEquals = function ifPlacementEquals(placement, value) {
  var fallback = arguments.length > 2 && arguments[2] !== undefined ? arguments[2] : 0;
  return function (props) {
    return props['data-placement'].split('-')[0] === placement ? value : fallback;
  };
};

var ArrowSpacing = 8;

var Arrow = _styledComponents["default"].div(_templateObject(), function (props) {
  return props.isVisible ? 'block' : 'none';
}, ifPlacementEquals('top', '0', ArrowSpacing), ifPlacementEquals('bottom', '0', ArrowSpacing), ifPlacementEquals('left', '0', ArrowSpacing), ifPlacementEquals('right', '0', ArrowSpacing), ifPlacementEquals('top', ArrowSpacing * -1, 'auto'), ifPlacementEquals('bottom', ArrowSpacing * -1, 'auto'), ifPlacementEquals('left', ArrowSpacing * -1, 'auto'), ifPlacementEquals('right', ArrowSpacing * -1, 'auto'), ifPlacementEquals('top', '0', ArrowSpacing), ifPlacementEquals('bottom', '0', ArrowSpacing), ifPlacementEquals('left', '0', ArrowSpacing), ifPlacementEquals('right', '0', ArrowSpacing), ifPlacementEquals('top', 'white', 'transparent'), ifPlacementEquals('bottom', 'white', 'transparent'), ifPlacementEquals('left', 'white', 'transparent'), ifPlacementEquals('right', 'white', 'transparent'));

var TooltipWrapper = _styledComponents["default"].div(_templateObject2(), function (props) {
  return props.hidden ? 'none' : 'inline-block';
}, function (props) {
  return props.zIndex;
}, function (props) {
  return !props.hasChrome && (0, _styledComponents.css)(_templateObject3(), ifPlacementEquals('top', 8), ifPlacementEquals('top-start', 8), ifPlacementEquals('bottom', 8), ifPlacementEquals('bottom-start', 8), ifPlacementEquals('right', 8), ifPlacementEquals('left', 8));
}, function (props) {
  return props.hasChrome && (0, _styledComponents.css)(_templateObject4(), ifPlacementEquals('top', ArrowSpacing + 2), ifPlacementEquals('bottom', ArrowSpacing + 2), ifPlacementEquals('right', ArrowSpacing + 2), ifPlacementEquals('left', ArrowSpacing + 2), _styles.spacing.borderRadius.small, _styles.typography.size.s1);
});

function Tooltip(_ref) {
  var placement = _ref.placement,
      hasChrome = _ref.hasChrome,
      children = _ref.children,
      arrowProps = _ref.arrowProps,
      tooltipRef = _ref.tooltipRef,
      arrowRef = _ref.arrowRef,
      zIndex = _ref.zIndex,
      props = (0, _objectWithoutProperties2["default"])(_ref, ["placement", "hasChrome", "children", "arrowProps", "tooltipRef", "arrowRef", "zIndex"]);
  return /*#__PURE__*/_react["default"].createElement(TooltipWrapper, (0, _extends2["default"])({
    hasChrome: hasChrome,
    "data-placement": placement,
    ref: tooltipRef,
    zIndex: zIndex
  }, props), /*#__PURE__*/_react["default"].createElement(Arrow, (0, _extends2["default"])({
    isVisible: hasChrome,
    "data-placement": placement,
    ref: arrowRef
  }, arrowProps)), children);
}

Tooltip.propTypes = {
  children: _propTypes["default"].node,
  hasChrome: _propTypes["default"].bool,

  /* eslint-disable-next-line */
  arrowProps: _propTypes["default"].any,
  placement: _propTypes["default"].string,
  arrowRef: _propTypes["default"].any,
  // eslint-disable-line react/forbid-prop-types
  tooltipRef: _propTypes["default"].any,
  // eslint-disable-line react/forbid-prop-types
  zIndex: _propTypes["default"].number
};
Tooltip.defaultProps = {
  children: undefined,
  hasChrome: true,
  placement: 'top',
  arrowProps: null,
  arrowRef: undefined,
  tooltipRef: undefined,
  zIndex: _styles.zIndex.tooltip
};
