// @storybook/design-system/dist/components/clipboard/ClipboardInput.stories.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Clipped = exports.Default = exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _ClipboardInput = require("./ClipboardInput");

var _default = {
  title: 'Clipboard/ClipboardInput'
};
exports["default"] = _default;

var Default = function Default() {
  return /*#__PURE__*/_react["default"].createElement("div", {
    style: {
      padding: '3em',
      width: 300
    }
  }, /*#__PURE__*/_react["default"].createElement(_ClipboardInput.ClipboardInput, {
    label: "Label",
    value: "https://www.chromatic.com"
  }));
};

exports.Default = Default;

var Clipped = function Clipped() {
  return /*#__PURE__*/_react["default"].createElement("div", {
    style: {
      padding: '3em',
      width: 250
    }
  }, /*#__PURE__*/_react["default"].createElement(_ClipboardInput.ClipboardInput, {
    label: "Label",
    value: "https://www.chromatic.com"
  }));
};

exports.Clipped = Clipped;
