// @storybook/design-system/dist/components/clipboard/ClipboardCode.stories.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Wrapped = exports.Default = exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _ClipboardCode = require("./ClipboardCode");

var _default = {
  title: 'Clipboard/ClipboardCode'
};
exports["default"] = _default;

var Default = function Default() {
  return /*#__PURE__*/_react["default"].createElement("div", {
    style: {
      padding: '3em',
      width: 300
    }
  }, /*#__PURE__*/_react["default"].createElement(_ClipboardCode.ClipboardCode, {
    code: "git checkout master"
  }));
};

exports.Default = Default;

var Wrapped = function Wrapped() {
  return /*#__PURE__*/_react["default"].createElement("div", {
    style: {
      padding: '3em',
      width: 250
    }
  }, /*#__PURE__*/_react["default"].createElement(_ClipboardCode.ClipboardCode, {
    code: "git checkout master"
  }));
};

exports.Wrapped = Wrapped;
