// @storybook/design-system/dist/components/clipboard/ClipboardInput.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ClipboardInput = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _taggedTemplateLiteral2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteral"));

var _react = _interopRequireDefault(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _Input = require("../Input");

var _ClipboardIcon = require("./ClipboardIcon");

function _templateObject3() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  position: absolute;\n  top: 4px;\n  right: 4px;\n"]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  width: 100%;\n  display: block;\n  && input {\n    padding-right: 26px !important;\n  }\n"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  position: relative;\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

var Container = _styledComponents["default"].div(_templateObject());

var StyledInput = (0, _styledComponents["default"])(_Input.Input)(_templateObject2());
var StyledClipboardIcon = (0, _styledComponents["default"])(_ClipboardIcon.ClipboardIcon)(_templateObject3());

var ClipboardInput = function ClipboardInput(_ref) {
  var value = _ref.value,
      props = (0, _objectWithoutProperties2["default"])(_ref, ["value"]);
  return /*#__PURE__*/_react["default"].createElement(Container, null, /*#__PURE__*/_react["default"].createElement(StyledInput, (0, _extends2["default"])({
    id: "clipboard-input"
  }, props, {
    icon: undefined,
    hideLabel: true,
    value: value,
    appearance: "code",
    readOnly: true
  })), /*#__PURE__*/_react["default"].createElement(StyledClipboardIcon, {
    toCopy: value
  }));
};

exports.ClipboardInput = ClipboardInput;
