// @storybook/design-system/dist/components/tag/TagList.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.TagList = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _taggedTemplateLiteral2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteral"));

var _react = _interopRequireWildcard(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _styles = require("../shared/styles");

var _Link = require("../Link");

var _TagItem = require("./TagItem");

function _templateObject2() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  font-size: ", "px;\n  padding-left: 5px;\n  padding-right: 5px;\n\n  &:focus {\n    outline: auto;\n  }\n"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  display: flex;\n  flex-wrap: wrap;\n\n  > * {\n    margin-right: 10px;\n    margin-bottom: 10px;\n  }\n\n  margin-bottom: -10px;\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

var TagListWrapper = _styledComponents["default"].div(_templateObject());

var MoreTagsButton = (0, _styledComponents["default"])(_Link.Link)(_templateObject2(), _styles.typography.size.s2);
var TagList = /*#__PURE__*/(0, _react.forwardRef)(function (_ref, ref) {
  var _ref$tags = _ref.tags,
      tags = _ref$tags === void 0 ? [] : _ref$tags,
      _ref$limit = _ref.limit,
      limit = _ref$limit === void 0 ? 4 : _ref$limit,
      _ref$isLoading = _ref.isLoading,
      isLoading = _ref$isLoading === void 0 ? false : _ref$isLoading,
      props = (0, _objectWithoutProperties2["default"])(_ref, ["tags", "limit", "isLoading"]);
  var primaryTags = tags.slice(0, limit);
  var moreTags = tags.slice(limit);

  var _useState = (0, _react.useState)(false),
      _useState2 = (0, _slicedToArray2["default"])(_useState, 2),
      moreTagsVisible = _useState2[0],
      setMoreTagsVisible = _useState2[1];

  var tagContent = /*#__PURE__*/_react["default"].createElement(_react["default"].Fragment, null, primaryTags, moreTagsVisible && moreTags, moreTags.length > 0 && !moreTagsVisible && /*#__PURE__*/_react["default"].createElement(MoreTagsButton, {
    isButton: true,
    appearance: "primary",
    onClick: function onClick() {
      return setMoreTagsVisible(true);
    }
  }, "+ ".concat(moreTags.length, " more")));

  return /*#__PURE__*/_react["default"].createElement(TagListWrapper, (0, _extends2["default"])({}, props, {
    ref: ref
  }), isLoading ? /*#__PURE__*/_react["default"].createElement(_react["default"].Fragment, null, /*#__PURE__*/_react["default"].createElement(_TagItem.TagItem, {
    isLoading: true
  }), /*#__PURE__*/_react["default"].createElement(_TagItem.TagItem, {
    isLoading: true
  }), /*#__PURE__*/_react["default"].createElement(_TagItem.TagItem, {
    isLoading: true
  }), /*#__PURE__*/_react["default"].createElement(_TagItem.TagItem, {
    isLoading: true
  })) : tagContent);
});
exports.TagList = TagList;
