// @storybook/design-system/dist/components/tag/TagLink.stories.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Loading = exports.WithLinkWrapper = exports.Default = exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _seedrandom = _interopRequireDefault(require("seedrandom"));

var _TagLink = require("./TagLink");

var _StoryLinkWrapper = require("../StoryLinkWrapper");

// @ts-ignore
// @ts-ignore
(0, _seedrandom["default"])('chromatic testing', {
  global: true
});
var _default = {
  component: _TagLink.TagLink,
  title: 'Tag/TagLink'
};
exports["default"] = _default;

var Default = function Default() {
  return /*#__PURE__*/_react["default"].createElement(_TagLink.TagLink, {
    href: "https://chromatic.com"
  }, "\u269B\uFE0F React");
};

exports.Default = Default;

var WithLinkWrapper = function WithLinkWrapper() {
  return /*#__PURE__*/_react["default"].createElement(_TagLink.TagLink, {
    to: "https://chromatic.com",
    LinkWrapper: _StoryLinkWrapper.StoryLinkWrapper
  }, "\u269B\uFE0F React");
};

exports.WithLinkWrapper = WithLinkWrapper;

var Loading = function Loading() {
  return /*#__PURE__*/_react["default"].createElement(_TagLink.TagLink, {
    isLoading: true
  });
};

exports.Loading = Loading;
