// @storybook/design-system/dist/components/table-of-contents/BulletLink.stories.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Series = exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _BulletLink = require("./BulletLink");

var _default = {
  title: 'TableOfContents/BulletLink',
  component: _BulletLink.BulletLink,
  parameters: {
    chromatic: {
      disable: true
    }
  }
};
exports["default"] = _default;
var currentPath = '/path-1'; // Bullet links should always be used in a series

var Series = function Series() {
  return /*#__PURE__*/_react["default"].createElement("ul", null, /*#__PURE__*/_react["default"].createElement(_BulletLink.BulletLink, {
    currentPath: currentPath,
    item: {
      path: '/path-1',
      title: 'Link 1'
    }
  }), /*#__PURE__*/_react["default"].createElement(_BulletLink.BulletLink, {
    currentPath: currentPath,
    item: {
      path: '/path-2',
      title: 'Link 2'
    }
  }));
};

exports.Series = Series;
