// @storybook/design-system/dist/components/table-of-contents/TableOfContentsItems.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.TableOfContentsItems = TableOfContentsItems;
exports.ItemType = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _taggedTemplateLiteral2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteral"));

var _react = _interopRequireDefault(require("react"));

var _styledComponents = _interopRequireWildcard(require("styled-components"));

var _BulletLink = require("./BulletLink");

var _ItemLink = require("./ItemLink");

var _MenuLink = require("./MenuLink");

var _styles = require("../shared/styles");

var _Icon = require("../Icon");

var _Link = require("../Link");

function _templateObject4() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n      > li {\n        padding-top: ", "px;\n\n        ul,\n        ol {\n          padding-top: 12px;\n          display: flex;\n          flex-direction: column;\n        }\n\n        > ul {\n          padding-left: 35px;\n\n          > li ul {\n            padding-left: 15px;\n          }\n        }\n\n        > ol {\n          padding-left: 20px;\n        }\n      }\n    "]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  list-style-type: none;\n  padding: 0;\n  margin: 0;\n\n  li {\n    padding-top: 12px;\n\n    &:first-child {\n      padding-top: 0;\n    }\n  }\n\n  ", "\n"]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  && {\n    width: 12px;\n    height: 12px;\n    color: ", ";\n    transform: translateY(1px) ", ";\n    ", "\n    bottom: -0.25em;\n  }\n"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  font-weight: ", ";\n  line-height: 24px;\n  word-break: break-word;\n  text-align: left;\n\n  > span {\n    display: flex;\n\n    svg {\n      flex: none;\n    }\n\n    span {\n      flex: 1;\n    }\n  }\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

var ItemType;
exports.ItemType = ItemType;

(function (ItemType) {
  ItemType["MENU"] = "menu";
  ItemType["LINK"] = "link";
  ItemType["BULLET_LINK"] = "bullet-link";
})(ItemType || (exports.ItemType = ItemType = {}));

var ItemComponent = function ItemComponent(_ref) {
  var currentPath = _ref.currentPath,
      item = _ref.item,
      isTopLevel = _ref.isTopLevel,
      setMenuOpenStateById = _ref.setMenuOpenStateById;

  if (item.type === ItemType.MENU) {
    return /*#__PURE__*/_react["default"].createElement(Menu, {
      currentPath: currentPath,
      item: item,
      isTopLevel: isTopLevel,
      setMenuOpenStateById: setMenuOpenStateById
    });
  }

  if (item.type === ItemType.LINK && item.path !== undefined) {
    return /*#__PURE__*/_react["default"].createElement(_ItemLink.ItemLink, {
      currentPath: currentPath,
      item: item
    });
  }

  if (item.type === ItemType.BULLET_LINK && item.path !== undefined) {
    return /*#__PURE__*/_react["default"].createElement(_BulletLink.BulletLink, {
      currentPath: currentPath,
      item: item
    });
  }

  return null;
};

var TopLevelMenuToggle = (0, _styledComponents["default"])(_Link.Link).attrs({
  isButton: true,
  tertiary: true
})(_templateObject(), _styles.typography.weight.bold);
var ArrowIcon = (0, _styledComponents["default"])(_Icon.Icon).attrs({
  icon: 'arrowright'
})(_templateObject2(), _styles.color.mediumdark, function (props) {
  return props.isOpen && "rotate(90deg)";
}, function (props) {
  return props.isTopLevel ? "margin-right: 8px;" : "margin-left: 8px;";
});

function Menu(_ref2) {
  var isTopLevel = _ref2.isTopLevel,
      item = _ref2.item,
      setMenuOpenStateById = _ref2.setMenuOpenStateById,
      currentPath = _ref2.currentPath,
      rest = (0, _objectWithoutProperties2["default"])(_ref2, ["isTopLevel", "item", "setMenuOpenStateById", "currentPath"]);
  if (!item.children) return null;

  var arrow = /*#__PURE__*/_react["default"].createElement(ArrowIcon, {
    isOpen: item.isOpen,
    isTopLevel: isTopLevel,
    "aria-hidden": true
  });

  var toggleOpenState = function toggleOpenState() {
    return setMenuOpenStateById({
      id: item.id,
      isOpen: !item.isOpen
    });
  };

  return /*#__PURE__*/_react["default"].createElement("li", null, isTopLevel ? /*#__PURE__*/_react["default"].createElement(TopLevelMenuToggle, {
    onClick: toggleOpenState
  }, arrow, /*#__PURE__*/_react["default"].createElement("span", null, item.title)) : /*#__PURE__*/_react["default"].createElement(_MenuLink.MenuLink, {
    isButton: true,
    onClick: toggleOpenState
  }, item.title, arrow), item.isOpen && /*#__PURE__*/_react["default"].createElement(TableOfContentsItems, (0, _extends2["default"])({
    items: item.children,
    isTopLevel: false,
    setMenuOpenStateById: setMenuOpenStateById,
    currentPath: currentPath
  }, rest)));
}

Menu.defaultProps = {
  isTopLevel: false
};

var List = _styledComponents["default"].ul(_templateObject3(), function (props) {
  return props.isTopLevel && (0, _styledComponents.css)(_templateObject4(), props.isFlatList ? 16 : 24);
});

function TableOfContentsItems(_ref3) {
  var className = _ref3.className,
      currentPath = _ref3.currentPath,
      _ref3$isTopLevel = _ref3.isTopLevel,
      isTopLevel = _ref3$isTopLevel === void 0 ? false : _ref3$isTopLevel,
      items = _ref3.items,
      setMenuOpenStateById = _ref3.setMenuOpenStateById,
      rest = (0, _objectWithoutProperties2["default"])(_ref3, ["className", "currentPath", "isTopLevel", "items", "setMenuOpenStateById"]);
  var isOrderedList = items.every(function (item) {
    return item.type === ItemType.BULLET_LINK;
  });
  var isFlatList = items.every(function (item) {
    return item.type === ItemType.LINK;
  });
  return /*#__PURE__*/_react["default"].createElement(List, {
    className: className,
    isTopLevel: isTopLevel,
    isFlatList: isFlatList,
    as: isOrderedList ? 'ol' : 'ul'
  }, items.map(function (item) {
    return /*#__PURE__*/_react["default"].createElement(ItemComponent, (0, _extends2["default"])({
      key: item.title,
      currentPath: currentPath,
      item: item,
      isTopLevel: isTopLevel,
      setMenuOpenStateById: setMenuOpenStateById
    }, rest));
  }));
}

TableOfContentsItems.defaultProps = {
  className: ''
};
