// @storybook/design-system/dist/components/table-of-contents/TableOfContents.stories.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.WithPathSwitcher = exports.WithOpenControls = exports.LinkWrappers = exports.NestedActivePath = exports.BasicNested = exports.BasicFlat = exports["default"] = void 0;

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _react = _interopRequireWildcard(require("react"));

var _TableOfContents = require("./TableOfContents");

var _TableOfContentsItems = require("./TableOfContentsItems");

var _StoryLinkWrapper = require("../StoryLinkWrapper");

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { (0, _defineProperty2["default"])(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

var _default = {
  title: 'TableOfContents',
  component: _TableOfContents.TableOfContents,
  decorators: [function (storyFn) {
    return /*#__PURE__*/_react["default"].createElement("div", {
      style: {
        width: 240,
        outline: '1px dotted grey'
      }
    }, storyFn());
  }]
};
exports["default"] = _default;
var items = [{
  title: 'Get Started',
  type: _TableOfContentsItems.ItemType.MENU,
  children: [{
    path: '/introduction',
    title: 'Introduction',
    type: _TableOfContentsItems.ItemType.BULLET_LINK
  }, {
    path: '/setup',
    title: 'Setup',
    type: _TableOfContentsItems.ItemType.BULLET_LINK
  }, {
    path: '/write-a-story',
    title: 'Write a story',
    type: _TableOfContentsItems.ItemType.BULLET_LINK
  }]
}, {
  title: 'ðŸ˜€ Customize',
  type: _TableOfContentsItems.ItemType.MENU,
  children: [{
    title: 'User interface',
    type: _TableOfContentsItems.ItemType.MENU,
    children: [{
      path: '/features-and-behavior',
      title: 'features and behavior dolor sit amet consectatur',
      type: _TableOfContentsItems.ItemType.LINK
    }, {
      path: '/theming',
      title: 'Theming',
      type: _TableOfContentsItems.ItemType.LINK
    }]
  }, {
    title: 'Head tags',
    type: _TableOfContentsItems.ItemType.LINK,
    path: '/head-tags'
  }, {
    title: 'Body tags',
    type: _TableOfContentsItems.ItemType.LINK,
    path: '/body-tags'
  }]
}, {
  title: 'âš™ï¸ Super long root heading that is very long',
  type: _TableOfContentsItems.ItemType.MENU,
  children: [{
    title: 'Verbose menu link that is extremely lengthy',
    type: _TableOfContentsItems.ItemType.MENU,
    children: [{
      path: '/lorem-ipsum',
      title: 'Lorem ipsum dolor sit amet consectatur',
      type: _TableOfContentsItems.ItemType.LINK
    }, {
      path: '/theming',
      title: 'Theming',
      type: _TableOfContentsItems.ItemType.LINK
    }]
  }, {
    title: 'Head tags',
    type: _TableOfContentsItems.ItemType.LINK,
    path: '/head-tags'
  }, {
    title: 'Body tags',
    type: _TableOfContentsItems.ItemType.LINK,
    path: '/body-tags'
  }]
}];

var findPaths = function findPaths(pathItems) {
  return (// @ts-ignore
    pathItems.flatMap(function (item) {
      return item.path || findPaths(item.children);
    })
  );
};

var paths = findPaths(items);

var BasicFlat = function BasicFlat(args) {
  return /*#__PURE__*/_react["default"].createElement(_TableOfContents.TableOfContents, args);
};

exports.BasicFlat = BasicFlat;
BasicFlat.args = {
  currentPath: '/essentials',
  items: [{
    title: 'â­ï¸  Popular',
    path: '/popular',
    type: _TableOfContentsItems.ItemType.LINK
  }, {
    title: 'ðŸ§©  Essentials',
    path: '/essentials',
    type: _TableOfContentsItems.ItemType.LINK
  }, {
    title: 'ðŸ›   Code',
    path: '/code',
    type: _TableOfContentsItems.ItemType.LINK
  }, {
    title: 'âš¡ï¸  Data & state',
    path: '/Data & state',
    type: _TableOfContentsItems.ItemType.LINK
  }, {
    title: 'ðŸ’…  Style',
    path: '/style',
    type: _TableOfContentsItems.ItemType.LINK
  }, {
    title: 'ðŸŽ¨  Design',
    path: '/design',
    type: _TableOfContentsItems.ItemType.LINK
  }, {
    title: 'âš™ï¸  Appearance',
    path: '/appearance',
    type: _TableOfContentsItems.ItemType.LINK
  }, {
    title: 'ðŸ—„  Organize',
    path: '/organize',
    type: _TableOfContentsItems.ItemType.LINK
  }]
};

var BasicNested = function BasicNested(args) {
  return /*#__PURE__*/_react["default"].createElement(_TableOfContents.TableOfContents, args);
};

exports.BasicNested = BasicNested;
BasicNested.args = {
  currentPath: paths[0],
  items: items
};
var NestedActivePath = BasicNested.bind({});
exports.NestedActivePath = NestedActivePath;
NestedActivePath.args = {
  currentPath: '/features-and-behavior',
  items: items
};

var addLinkWrappers = function addLinkWrappers(itemsToCompose) {
  return itemsToCompose.map(function (item) {
    if (item.type === _TableOfContentsItems.ItemType.LINK || item.type === _TableOfContentsItems.ItemType.BULLET_LINK) return _objectSpread(_objectSpread({}, item), {}, {
      LinkWrapper: _StoryLinkWrapper.StoryLinkWrapper
    });
    if (item.children) return _objectSpread(_objectSpread({}, item), {}, {
      children: addLinkWrappers(item.children)
    });
    return item;
  });
};

var itemsWithLinkWrappers = addLinkWrappers(items);
var LinkWrappers = BasicNested.bind({});
exports.LinkWrappers = LinkWrappers;
LinkWrappers.args = {
  currentPath: paths[0],
  items: itemsWithLinkWrappers
};

var WithOpenControls = function WithOpenControls() {
  return /*#__PURE__*/_react["default"].createElement(_TableOfContents.TableOfContents, {
    currentPath: paths[0],
    items: items
  }, function (_ref) {
    var menu = _ref.menu,
        toggleAllOpen = _ref.toggleAllOpen,
        toggleAllClosed = _ref.toggleAllClosed;
    return /*#__PURE__*/_react["default"].createElement(_react["default"].Fragment, null, /*#__PURE__*/_react["default"].createElement("div", {
      style: {
        marginBottom: '10px'
      }
    }, /*#__PURE__*/_react["default"].createElement("button", {
      onClick: toggleAllOpen
    }, "Open all"), /*#__PURE__*/_react["default"].createElement("button", {
      onClick: toggleAllClosed
    }, "Close all")), menu);
  });
};

exports.WithOpenControls = WithOpenControls;

var WithPathSwitcher = function WithPathSwitcher() {
  var _useState = (0, _react.useState)(paths[0]),
      _useState2 = (0, _slicedToArray2["default"])(_useState, 2),
      currentPath = _useState2[0],
      setCurrentPath = _useState2[1];

  var cyclePath = function cyclePath() {
    var currentIndex = paths.indexOf(currentPath);
    setCurrentPath(paths[currentIndex + 1] || paths[0]);
  };

  return /*#__PURE__*/_react["default"].createElement(_react["default"].Fragment, null, /*#__PURE__*/_react["default"].createElement("button", {
    style: {
      marginBottom: '10px'
    },
    onClick: cyclePath
  }, "Change path"), /*#__PURE__*/_react["default"].createElement(_TableOfContents.TableOfContents, {
    currentPath: currentPath,
    items: items
  }));
};

exports.WithPathSwitcher = WithPathSwitcher;
