// @storybook/design-system/dist/components/Avatar.stories.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Organization = exports.Tiny = exports.Small = exports.Medium = exports.Large = exports["default"] = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireDefault(require("react"));

var _Avatar = require("./Avatar");

var _default = {
  title: 'Avatar',
  component: _Avatar.Avatar,
  args: {
    type: 'user'
  }
};
exports["default"] = _default;

var Base = function Base(_ref) {
  var src = _ref.src,
      props = (0, _objectWithoutProperties2["default"])(_ref, ["src"]);
  return /*#__PURE__*/_react["default"].createElement("div", null, /*#__PURE__*/_react["default"].createElement(_Avatar.Avatar, (0, _extends2["default"])({
    isLoading: true
  }, props)), /*#__PURE__*/_react["default"].createElement(_Avatar.Avatar, (0, _extends2["default"])({
    username: "Tom Coleman"
  }, props)), /*#__PURE__*/_react["default"].createElement(_Avatar.Avatar, (0, _extends2["default"])({
    username: "Tom Coleman",
    src: src || 'https://avatars2.githubusercontent.com/u/132554'
  }, props)));
};

var Large = function Large() {
  return /*#__PURE__*/_react["default"].createElement(Base, {
    size: "large"
  });
};

exports.Large = Large;

var Medium = function Medium() {
  return /*#__PURE__*/_react["default"].createElement(Base, null);
};

exports.Medium = Medium;

var Small = function Small() {
  return /*#__PURE__*/_react["default"].createElement(Base, {
    size: "small"
  });
};

exports.Small = Small;

var Tiny = function Tiny() {
  return /*#__PURE__*/_react["default"].createElement(Base, {
    size: "tiny"
  });
};

exports.Tiny = Tiny;

var Organization = function Organization() {
  return /*#__PURE__*/_react["default"].createElement(Base, {
    type: "organization",
    username: "Chromatic",
    src: "/chromatic-logo-square.png"
  });
};

exports.Organization = Organization;
