// @storybook/design-system/dist/components/Badge.stories.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.WithIcon = exports.All = exports.Basic = exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _Badge = require("./Badge");

var _Icon = require("./Icon");

var _default = {
  title: 'Badge',
  component: _Badge.Badge
};
exports["default"] = _default;

var Basic = function Basic(args) {
  return /*#__PURE__*/_react["default"].createElement(_Badge.Badge, args, "Badge");
};

exports.Basic = Basic;

var All = function All() {
  return /*#__PURE__*/_react["default"].createElement("div", null, /*#__PURE__*/_react["default"].createElement(_Badge.Badge, {
    status: "positive"
  }, "Positive"), /*#__PURE__*/_react["default"].createElement(_Badge.Badge, {
    status: "negative"
  }, "Negative"), /*#__PURE__*/_react["default"].createElement(_Badge.Badge, {
    status: "neutral"
  }, "Neutral"), /*#__PURE__*/_react["default"].createElement(_Badge.Badge, {
    status: "error"
  }, "Error"), /*#__PURE__*/_react["default"].createElement(_Badge.Badge, {
    status: "warning"
  }, "Warning"));
};

exports.All = All;

var WithIcon = function WithIcon() {
  return /*#__PURE__*/_react["default"].createElement(_Badge.Badge, {
    status: "warning"
  }, /*#__PURE__*/_react["default"].createElement(_Icon.Icon, {
    icon: "check",
    inline: true
  }), "with icon");
};

exports.WithIcon = WithIcon;
