// @storybook/design-system/dist/components/ButtonToggle.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ButtonToggle = ButtonToggle;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _taggedTemplateLiteral2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteral"));

var _polished = require("polished");

var _react = _interopRequireDefault(require("react"));

var _styledComponents = _interopRequireWildcard(require("styled-components"));

var _styles = require("./shared/styles");

var _Link = require("./Link");

var _TooltipNote = require("./tooltip/TooltipNote");

var _WithTooltip = _interopRequireDefault(require("./tooltip/WithTooltip"));

function _templateObject10() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n      background-color: ", ";\n      border-radius: 3em;\n    "]);

  _templateObject10 = function _templateObject10() {
    return data;
  };

  return data;
}

function _templateObject9() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  display: inline-flex;\n  align-items: center;\n  white-space: nowrap;\n\n  ", "\n"]);

  _templateObject9 = function _templateObject9() {
    return data;
  };

  return data;
}

function _templateObject8() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n      color: ", ";\n      box-shadow: ", " 0 -3px 0 0 inset;\n    "]);

  _templateObject8 = function _templateObject8() {
    return data;
  };

  return data;
}

function _templateObject7() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  font-size: ", "px;\n  font-weight: ", ";\n  line-height: 20px;\n  display: inline-block;\n  padding: 10px 15px;\n  &:hover {\n    color: ", ";\n  }\n  ", ";\n"]);

  _templateObject7 = function _templateObject7() {
    return data;
  };

  return data;
}

function _templateObject6() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n      background: #e3f3ff;\n      color: ", ";\n      img {\n        opacity: 1;\n      }\n    "]);

  _templateObject6 = function _templateObject6() {
    return data;
  };

  return data;
}

function _templateObject5() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  font-size: ", "px;\n  font-weight: ", ";\n  line-height: 1;\n  display: inline-block;\n  vertical-align: top;\n  padding: 4px 8px;\n  border-radius: 10px;\n  &:hover {\n    background: #e3f3ff;\n    color: ", ";\n    img {\n      opacity: 1;\n    }\n  }\n  img {\n    height: 1rem;\n    width: 1rem;\n    opacity: 0.3;\n    transition: all 150ms ease-out;\n    margin: -3px 0;\n  }\n  ", ";\n"]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n      &:first-child {\n        margin-right: -", "px;\n      }\n      :not(:first-child):not(:last-child) {\n        margin-left: -", "px;\n        margin-right: -", "px;\n      }\n      &:last-child {\n        margin-left: -", "px;\n      }\n    "]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  ", "\n\n  &:first-child ", " {\n    padding-left: ", "px;\n  }\n\n  &:last-child ", " {\n    padding-right: ", "px;\n  }\n\n  &:hover ", " {\n    position: relative;\n    z-index: 1;\n  }\n"]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n      background-color: ", ";\n      box-shadow: ", " 0 0 0 1px inset;\n      color: ", ";\n      padding-right: ", "px;\n      padding-left: ", "px;\n    "]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  border: 0;\n  border-radius: 3em;\n  cursor: pointer;\n  display: inline-block;\n  overflow: hidden;\n  padding: 8px ", "px;\n  position: relative;\n  text-align: center;\n  text-decoration: none;\n  transition: background-color 150ms ease-out, color 150ms ease-out;\n  transform: translate3d(0, 0, 0);\n  vertical-align: top;\n  white-space: nowrap;\n  user-select: none;\n  opacity: 1;\n  margin: 0;\n\n  font-size: ", "px;\n  font-weight: ", ";\n  line-height: 1;\n\n  color: ", ";\n  background-color: transparent;\n\n  ", "\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

var activePadding = 20;
var inActivePadding = 15;

var StyledButton = _styledComponents["default"].button(_templateObject(), inActivePadding, _styles.typography.size.s1, _styles.typography.weight.extrabold, _styles.color.dark, function (props) {
  return props.isActive && (0, _styledComponents.css)(_templateObject2(), _styles.color.lightest, (0, _polished.opacify)(0.05, _styles.color.border), _styles.color.darkest, activePadding, activePadding);
});

var TooltipWrapper = (0, _styledComponents["default"])(_WithTooltip["default"])(_templateObject3(), function (props) {
  return props.isActive && (0, _styledComponents.css)(_templateObject4(), activePadding - inActivePadding, activePadding - inActivePadding, activePadding - inActivePadding, activePadding - inActivePadding);
}, StyledButton, activePadding, StyledButton, activePadding, StyledButton);
var PillButton = (0, _styledComponents["default"])(_Link.Link)(_templateObject5(), _styles.typography.size.s1, _styles.typography.weight.bold, _styles.color.secondary, function (props) {
  return props.active && (0, _styledComponents.css)(_templateObject6(), _styles.color.secondary);
});
var TabButton = (0, _styledComponents["default"])(_Link.Link)(_templateObject7(), _styles.typography.size.s2 - 1, _styles.typography.weight.bold, _styles.color.secondary, function (props) {
  return props.active && (0, _styledComponents.css)(_templateObject8(), _styles.color.secondary, _styles.color.secondary);
});

var Wrapper = _styledComponents["default"].div(_templateObject9(), function (props) {
  return props.appearance === 'outline' && (0, _styledComponents.css)(_templateObject10(), (0, _polished.opacify)(0.05, _styles.color.border));
});

function ButtonToggle(_ref) {
  var titles = _ref.titles,
      onSelectIndex = _ref.onSelectIndex,
      selectedIndex = _ref.selectedIndex,
      appearance = _ref.appearance,
      props = (0, _objectWithoutProperties2["default"])(_ref, ["titles", "onSelectIndex", "selectedIndex", "appearance"]);
  return /*#__PURE__*/_react["default"].createElement(Wrapper, (0, _extends2["default"])({
    appearance: appearance
  }, props), titles.map(function (_ref2, index) {
    var title = _ref2.title,
        tooltip = _ref2.tooltip;

    switch (appearance) {
      case 'pill':
        return /*#__PURE__*/_react["default"].createElement(PillButton // eslint-disable-next-line react/no-array-index-key
        , {
          key: index,
          active: index === selectedIndex,
          onClick: function onClick() {
            return onSelectIndex(index);
          },
          title: tooltip,
          secondary: true,
          isButton: true
        }, title);

      case 'tab':
        return /*#__PURE__*/_react["default"].createElement(TabButton // eslint-disable-next-line react/no-array-index-key
        , {
          key: index,
          active: index === selectedIndex,
          onClick: function onClick() {
            return onSelectIndex(index);
          },
          title: tooltip,
          secondary: true,
          isButton: true
        }, title);

      case 'outline':
        return /*#__PURE__*/_react["default"].createElement(TooltipWrapper, {
          tagName: "span",
          key: typeof title === 'string' ? title : index,
          hasChrome: false,
          placement: "bottom",
          trigger: "hover",
          tooltip: /*#__PURE__*/_react["default"].createElement(_TooltipNote.TooltipNote, {
            note: tooltip
          }),
          isActive: index === selectedIndex
        }, /*#__PURE__*/_react["default"].createElement(StyledButton, {
          isActive: index === selectedIndex,
          onClick: function onClick() {
            return onSelectIndex(index);
          }
        }, title));

      default:
        return 'this appearance is not supported';
    }
  }));
}

ButtonToggle.defaultProps = {
  appearance: 'outline'
};
