// @storybook/design-system/dist/components/Button.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Button = exports.StyledButton = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _taggedTemplateLiteral2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteral"));

var _react = _interopRequireWildcard(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _polished = require("polished");

var _styles = require("./shared/styles");

var _animation = require("./shared/animation");

function _templateObject4() {
  var data = (0, _taggedTemplateLiteral2["default"])([""]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  border: 0;\n  border-radius: 3em;\n  cursor: pointer;\n  display: inline-block;\n  overflow: hidden;\n  padding: ", ";\n  position: relative;\n  text-align: center;\n  text-decoration: none;\n  transition: all 150ms ease-out;\n  transform: translate3d(0, 0, 0);\n  will-change: transform;\n  vertical-align: top;\n  white-space: nowrap;\n  user-select: none;\n  opacity: 1;\n  margin: 0;\n  background: transparent;\n\n  font-size: ", "px;\n  font-weight: ", ";\n  line-height: 1;\n\n  ", "\n\n  ", " {\n    transform: scale3d(1,1,1) translate3d(0,0,0);\n    transition: transform 700ms ", ";\n    opacity: 1;\n  }\n\n  ", " {\n    transform: translate3d(0, 100%, 0);\n  }\n\n  svg {\n    height: ", "px;\n    width: ", "px;\n    vertical-align: top;\n    margin-right: ", "px;\n    margin-top: ", "px;\n    margin-bottom: ", "px;\n\n    /* Necessary for js mouse events to not glitch out when hovering on svgs */\n    pointer-events: none;\n  }\n\n  ", "\n\n  ", "\n\n  ", "\n\n  ", "\n\n  ", "\n\n  ", "\n\n  ", "\n\n  ", ";\n\n  ", ";\n\n  ", ";\n\n    ", "\n\n    ", "\n\n    ", ";\n"]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  position: absolute;\n  top: 50%;\n  left: 0;\n  right: 0;\n  opacity: 0;\n"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  display: inline-block;\n  vertical-align: top;\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

var Text = _styledComponents["default"].span(_templateObject());

var Loading = _styledComponents["default"].span(_templateObject2());

var APPEARANCES = {
  PRIMARY: 'primary',
  PRIMARY_OUTLINE: 'primaryOutline',
  SECONDARY: 'secondary',
  SECONDARY_OUTLINE: 'secondaryOutline',
  TERTIARY: 'tertiary',
  OUTLINE: 'outline',
  INVERSE_PRIMARY: 'inversePrimary',
  INVERSE_SECONDARY: 'inverseSecondary',
  INVERSE_OUTLINE: 'inverseOutline'
};
var SIZES = {
  SMALL: 'small',
  MEDIUM: 'medium'
};

var StyledButton = _styledComponents["default"].button(_templateObject3(), function (props) {
  return props.size === SIZES.SMALL ? '8px 16px' : '13px 20px';
}, function (props) {
  return props.size === SIZES.SMALL ? _styles.typography.size.s1 : _styles.typography.size.s2;
}, _styles.typography.weight.extrabold, function (props) {
  return !props.isLoading && "\n      &:hover {\n        transform: translate3d(0, -2px, 0);\n        box-shadow: rgba(0, 0, 0, 0.2) 0 2px 6px 0;\n      }\n\n      &:active {\n        transform: translate3d(0, 0, 0);\n      }\n    ";
}, Text, _animation.easing.rubber, Loading, function (props) {
  return props.size === SIZES.SMALL ? '14' : '16';
}, function (props) {
  return props.size === SIZES.SMALL ? '14' : '16';
}, function (props) {
  return props.size === SIZES.SMALL ? '4' : '6';
}, function (props) {
  return props.size === SIZES.SMALL ? '-1' : '-2';
}, function (props) {
  return props.size === SIZES.SMALL ? '-1' : '-2';
}, function (props) {
  return props.disabled && "\n      cursor: not-allowed !important;\n      opacity: 0.5;\n      &:hover {\n        transform: none;\n      }\n    ";
}, function (props) {
  return props.isUnclickable && "\n      cursor: default !important;\n      pointer-events: none;\n      &:hover {\n        transform: none;\n      }\n    ";
}, function (props) {
  return props.isLoading && "\n      cursor: progress !important;\n      opacity: 0.7;\n\n      ".concat(Loading, " {\n        transition: transform 700ms ").concat(_animation.easing.rubber, ";\n        transform: translate3d(0, -50%, 0);\n        opacity: 1;\n      }\n\n      ").concat(Text, " {\n        transform: scale3d(0, 0, 1) translate3d(0, -100%, 0);\n        opacity: 0;\n      }\n\n      &:hover {\n        transform: none;\n      }\n    ");
}, function (props) {
  return props.containsIcon && "\n      svg {\n        display: block;\n        margin: 0;\n      }\n      padding: ".concat(props.size === SIZES.SMALL ? '7' : '12', "px;\n    ");
}, function (props) {
  return props.appearance === APPEARANCES.PRIMARY && "\n      background: ".concat(_styles.color.primary, ";\n      color: ").concat(_styles.color.lightest, ";\n\n      ").concat(!props.isLoading && "\n          &:hover {\n            background: ".concat((0, _polished.darken)(0.05, _styles.color.primary), ";\n          }\n          &:active {\n            box-shadow: rgba(0, 0, 0, 0.1) 0 0 0 3em inset;\n          }\n        "), "\n    ");
}, function (props) {
  return props.appearance === APPEARANCES.SECONDARY && "\n      background: ".concat(_styles.color.secondary, ";\n      color: ").concat(_styles.color.lightest, ";\n\n      ").concat(!props.isLoading && "\n          &:hover {\n            background: ".concat((0, _polished.darken)(0.05, _styles.color.secondary), ";\n          }\n          &:active {\n            box-shadow: rgba(0, 0, 0, 0.1) 0 0 0 3em inset;\n          }\n        "), "\n    ");
}, function (props) {
  return props.appearance === APPEARANCES.TERTIARY && "\n      background: ".concat(_styles.color.tertiary, ";\n      color: ").concat(_styles.color.darkest, ";\n\n      ").concat(!props.isLoading && "\n          &:hover {\n            background: ".concat((0, _polished.darken)(0.05, _styles.color.tertiary), ";\n          }\n          &:active {\n            box-shadow: rgba(0, 0, 0, 0.1) 0 0 0 3em inset;\n          }\n        "), "\n    ");
}, function (props) {
  return props.appearance === APPEARANCES.OUTLINE && "\n      box-shadow: ".concat((0, _polished.opacify)(0.05, _styles.color.border), " 0 0 0 1px inset;\n      color: ").concat(_styles.color.dark, ";\n      background: transparent;\n\n      ").concat(!props.isLoading && "\n          &:hover {\n            box-shadow: ".concat((0, _polished.opacify)(0.3, _styles.color.border), " 0 0 0 1px inset;\n          }\n\n          &:active {\n            background: ").concat((0, _polished.opacify)(0.05, _styles.color.border), ";\n            box-shadow: transparent 0 0 0 1px inset;\n            color: ").concat(_styles.color.darkest, ";\n          }\n\n          &:active:focus:hover {\n            ", "\n            background: ").concat((0, _polished.opacify)(0.05, _styles.color.border), ";\n            box-shadow: transparent 0 0 0 1px inset;\n          }\n        "), ";\n    ");
}, function (props) {
  return props.appearance === APPEARANCES.PRIMARY_OUTLINE && "\n      box-shadow: ".concat(_styles.color.primary, " 0 0 0 1px inset;\n      color: ").concat(_styles.color.primary, ";\n\n      &:hover {\n        box-shadow: ").concat(_styles.color.primary, " 0 0 0 1px inset;\n        background: transparent;\n      }\n\n      &:active {\n        background: ").concat(_styles.color.primary, ";\n        box-shadow: ").concat(_styles.color.primary, " 0 0 0 1px inset;\n        color: ").concat(_styles.color.lightest, ";\n      }\n    ");
}, function (props) {
  return props.appearance === APPEARANCES.SECONDARY_OUTLINE && "\n      box-shadow: ".concat(_styles.color.secondary, " 0 0 0 1px inset;\n      color: ").concat(_styles.color.secondary, ";\n\n      &:hover {\n        box-shadow: ").concat(_styles.color.secondary, " 0 0 0 1px inset;\n        background: transparent;\n      }\n\n      &:active {\n        background: ").concat(_styles.color.secondary, ";\n        box-shadow: ").concat(_styles.color.secondary, " 0 0 0 1px inset;\n        color: ").concat(_styles.color.lightest, ";\n      }\n    ");
}, function (props) {
  return props.appearance === APPEARANCES.INVERSE_PRIMARY && "\n        background: ".concat(_styles.color.lightest, ";\n        color: ").concat(_styles.color.primary, ";\n\n        ").concat(!props.isLoading && "\n            &:hover {\n              background: ".concat(_styles.color.lightest, ";\n            }\n            &:active {\n              box-shadow: rgba(0, 0, 0, 0.1) 0 0 0 3em inset;\n            }\n        "), "\n    ");
}, function (props) {
  return props.appearance === APPEARANCES.INVERSE_SECONDARY && "\n        background: ".concat(_styles.color.lightest, ";\n        color: ").concat(_styles.color.secondary, ";\n\n        ").concat(!props.isLoading && "\n            &:hover {\n              background: ".concat(_styles.color.lightest, ";\n            }\n            &:active {\n              box-shadow: rgba(0, 0, 0, 0.1) 0 0 0 3em inset;\n            }\n        "), "\n    ");
}, function (props) {
  return props.appearance === APPEARANCES.INVERSE_OUTLINE && "\n        box-shadow: ".concat(_styles.color.lightest, " 0 0 0 1px inset;\n        color: ").concat(_styles.color.lightest, ";\n\n        &:hover {\n          box-shadow: ").concat(_styles.color.lightest, " 0 0 0 1px inset;\n          background: transparent;\n        }\n\n        &:active {\n          background: ").concat(_styles.color.lightest, ";\n          box-shadow: ").concat(_styles.color.lightest, " 0 0 0 1px inset;\n          color: ").concat(_styles.color.darkest, ";\n        }\n    ");
});

exports.StyledButton = StyledButton;

var ButtonLink = _styledComponents["default"].a(_templateObject4());

var Button = /*#__PURE__*/(0, _react.forwardRef)(function (_ref, ref) {
  var children = _ref.children,
      _ref$isDisabled = _ref.isDisabled,
      isDisabled = _ref$isDisabled === void 0 ? false : _ref$isDisabled,
      isLoading = _ref.isLoading,
      _ref$loadingText = _ref.loadingText,
      loadingText = _ref$loadingText === void 0 ? null : _ref$loadingText,
      isLink = _ref.isLink,
      _ref$ButtonWrapper = _ref.ButtonWrapper,
      ButtonWrapper = _ref$ButtonWrapper === void 0 ? null : _ref$ButtonWrapper,
      _ref$appearance = _ref.appearance,
      appearance = _ref$appearance === void 0 ? 'tertiary' : _ref$appearance,
      rest = (0, _objectWithoutProperties2["default"])(_ref, ["children", "isDisabled", "isLoading", "loadingText", "isLink", "ButtonWrapper", "appearance"]);

  if (ButtonWrapper) {
    return /*#__PURE__*/_react["default"].createElement(StyledButton, (0, _extends2["default"])({
      as: ButtonWrapper,
      disabled: isDisabled,
      isLoading: isLoading,
      appearance: appearance
    }, rest, {
      ref: ref
    }), /*#__PURE__*/_react["default"].createElement(_react["default"].Fragment, null, /*#__PURE__*/_react["default"].createElement(Text, null, children), isLoading && /*#__PURE__*/_react["default"].createElement(Loading, null, loadingText || 'Loading...')));
  }

  if (isLink) {
    return /*#__PURE__*/_react["default"].createElement(StyledButton, (0, _extends2["default"])({
      as: ButtonLink,
      isLoading: isLoading,
      appearance: appearance
    }, rest, {
      ref: ref
    }), /*#__PURE__*/_react["default"].createElement(_react["default"].Fragment, null, /*#__PURE__*/_react["default"].createElement(Text, null, children), isLoading && /*#__PURE__*/_react["default"].createElement(Loading, null, loadingText || 'Loading...')));
  }

  return /*#__PURE__*/_react["default"].createElement(StyledButton, (0, _extends2["default"])({
    disabled: isDisabled,
    isLoading: isLoading,
    appearance: appearance
  }, rest, {
    ref: ref
  }), /*#__PURE__*/_react["default"].createElement(_react["default"].Fragment, null, /*#__PURE__*/_react["default"].createElement(Text, null, children), isLoading && /*#__PURE__*/_react["default"].createElement(Loading, null, loadingText || 'Loading...')));
});
exports.Button = Button;
