// @storybook/design-system/dist/components/Button.stories.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.AnchorWrapper = exports.BasicWrapper = exports.ContainsIcon = exports.Disabled = exports.Loading = exports.Sizes = exports.All = exports.Basic = exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _addonActions = require("@storybook/addon-actions");

var _Button = require("./Button");

var _Icon = require("./Icon");

var _StoryLinkWrapper = require("./StoryLinkWrapper");

var _default = {
  title: 'Button',
  component: _Button.Button
};
exports["default"] = _default;

var Basic = function Basic(args) {
  return /*#__PURE__*/_react["default"].createElement(_Button.Button, args);
};

exports.Basic = Basic;
Basic.args = {
  children: 'Label'
};

var All = function All() {
  return /*#__PURE__*/_react["default"].createElement(_react["default"].Fragment, null, /*#__PURE__*/_react["default"].createElement(_Button.Button, {
    appearance: "primary"
  }, "Primary"), /*#__PURE__*/_react["default"].createElement(_Button.Button, {
    appearance: "secondary"
  }, "Secondary"), /*#__PURE__*/_react["default"].createElement(_Button.Button, {
    appearance: "tertiary"
  }, "Tertiary"), /*#__PURE__*/_react["default"].createElement(_Button.Button, {
    appearance: "outline"
  }, "Outline"), /*#__PURE__*/_react["default"].createElement(_Button.Button, {
    appearance: "primaryOutline"
  }, "Outline primary"), /*#__PURE__*/_react["default"].createElement(_Button.Button, {
    appearance: "secondaryOutline"
  }, "Outline secondary"), /*#__PURE__*/_react["default"].createElement("div", {
    style: {
      background: 'grey',
      display: 'inline-block'
    }
  }, /*#__PURE__*/_react["default"].createElement(_Button.Button, {
    appearance: "inversePrimary"
  }, "Primary inverse")), /*#__PURE__*/_react["default"].createElement("div", {
    style: {
      background: 'grey',
      display: 'inline-block'
    }
  }, /*#__PURE__*/_react["default"].createElement(_Button.Button, {
    appearance: "inverseSecondary"
  }, "Secondary inverse")), /*#__PURE__*/_react["default"].createElement("div", {
    style: {
      background: 'grey',
      display: 'inline-block'
    }
  }, /*#__PURE__*/_react["default"].createElement(_Button.Button, {
    appearance: "inverseOutline"
  }, "Outline inverse")));
};

exports.All = All;

var Sizes = function Sizes() {
  return /*#__PURE__*/_react["default"].createElement(_react["default"].Fragment, null, /*#__PURE__*/_react["default"].createElement(_Button.Button, {
    appearance: "primary"
  }, "Default"), /*#__PURE__*/_react["default"].createElement(_Button.Button, {
    appearance: "primary",
    size: "small"
  }, "Small"));
};

exports.Sizes = Sizes;

var Loading = function Loading() {
  return /*#__PURE__*/_react["default"].createElement(_react["default"].Fragment, null, /*#__PURE__*/_react["default"].createElement(_Button.Button, {
    appearance: "primary",
    isLoading: true
  }, "Primary"), /*#__PURE__*/_react["default"].createElement(_Button.Button, {
    appearance: "secondary",
    isLoading: true
  }, "Secondary"), /*#__PURE__*/_react["default"].createElement(_Button.Button, {
    appearance: "tertiary",
    isLoading: true
  }, "Tertiary"), /*#__PURE__*/_react["default"].createElement(_Button.Button, {
    appearance: "outline",
    isLoading: true
  }, "Outline"), /*#__PURE__*/_react["default"].createElement(_Button.Button, {
    appearance: "outline",
    isLoading: true,
    loadingText: "Custom..."
  }, "Outline"));
};

exports.Loading = Loading;

var Disabled = function Disabled() {
  return /*#__PURE__*/_react["default"].createElement(_react["default"].Fragment, null, /*#__PURE__*/_react["default"].createElement(_Button.Button, {
    appearance: "primary",
    isDisabled: true
  }, "Primary"), /*#__PURE__*/_react["default"].createElement(_Button.Button, {
    appearance: "secondary",
    isDisabled: true
  }, "Secondary"), /*#__PURE__*/_react["default"].createElement(_Button.Button, {
    appearance: "tertiary",
    isDisabled: true
  }, "Tertiary"), /*#__PURE__*/_react["default"].createElement(_Button.Button, {
    appearance: "outline",
    isDisabled: true
  }, "Outline"));
};

exports.Disabled = Disabled;

var ContainsIcon = function ContainsIcon() {
  return /*#__PURE__*/_react["default"].createElement(_react["default"].Fragment, null, /*#__PURE__*/_react["default"].createElement(_Button.Button, {
    appearance: "outline",
    containsIcon: true
  }, /*#__PURE__*/_react["default"].createElement(_Icon.Icon, {
    icon: "link",
    "aria-label": "Link"
  })), /*#__PURE__*/_react["default"].createElement(_Button.Button, {
    appearance: "outline",
    size: "small",
    containsIcon: true
  }, /*#__PURE__*/_react["default"].createElement(_Icon.Icon, {
    icon: "link",
    "aria-label": "Link"
  })));
}; // eslint-disable-next-line react/button-has-type


exports.ContainsIcon = ContainsIcon;

var ButtonWrapper = function ButtonWrapper(props) {
  return /*#__PURE__*/_react["default"].createElement("button", props);
};

var BasicWrapper = function BasicWrapper() {
  return /*#__PURE__*/_react["default"].createElement(_Button.Button, {
    ButtonWrapper: ButtonWrapper,
    appearance: "primary",
    onClick: (0, _addonActions.action)('button action click')
  }, "Button that passes clicks through");
};

exports.BasicWrapper = BasicWrapper;

var AnchorWrapper = function AnchorWrapper() {
  return /*#__PURE__*/_react["default"].createElement(_Button.Button, {
    ButtonWrapper: _StoryLinkWrapper.StoryLinkWrapper,
    appearance: "primary",
    href: "http://storybook.js.org"
  }, "Button that passes href through");
};

exports.AnchorWrapper = AnchorWrapper;
