// @storybook/design-system/dist/components/ButtonToggle.stories.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Tab = exports.PillWImage = exports.PillWText = exports.OutlineThreeButtonsSmall = exports.OutlineButtonsSmall = exports["default"] = void 0;

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var _addonActions = require("@storybook/addon-actions");

var _react = _interopRequireWildcard(require("react"));

var _ButtonToggle = require("./ButtonToggle");

var _chrome = _interopRequireDefault(require("../images/chrome.svg"));

var _firefox = _interopRequireDefault(require("../images/firefox.svg"));

var _ie = _interopRequireDefault(require("../images/ie.svg"));

var _safari = _interopRequireDefault(require("../images/safari.svg"));

// @ts-ignore
// @ts-ignore
// @ts-ignore
// @ts-ignore
var onSelectIndex = (0, _addonActions.action)('onSelectIndex');
var _default = {
  title: 'ButtonToggle'
};
exports["default"] = _default;

var OutlineButtonsSmall = function OutlineButtonsSmall() {
  var _useState = (0, _react.useState)(1),
      _useState2 = (0, _slicedToArray2["default"])(_useState, 2),
      index = _useState2[0],
      setIndex = _useState2[1];

  return /*#__PURE__*/_react["default"].createElement(_ButtonToggle.ButtonToggle, {
    titles: [{
      title: '1 up',
      tooltip: '1 up [1]'
    }, {
      title: '2 up',
      tooltip: '2 up [2]'
    }],
    selectedIndex: index,
    onSelectIndex: function onSelectIndex(idx) {
      return setIndex(idx);
    }
  });
};

exports.OutlineButtonsSmall = OutlineButtonsSmall;
OutlineButtonsSmall.story = {
  name: 'outline buttons small'
};

var OutlineThreeButtonsSmall = function OutlineThreeButtonsSmall() {
  var _useState3 = (0, _react.useState)(1),
      _useState4 = (0, _slicedToArray2["default"])(_useState3, 2),
      index = _useState4[0],
      setIndex = _useState4[1];

  return /*#__PURE__*/_react["default"].createElement(_ButtonToggle.ButtonToggle, {
    titles: [{
      title: '1 up',
      tooltip: '1 up [1]'
    }, {
      title: '2 up',
      tooltip: '2 up [2]'
    }, {
      title: 'Diff',
      tooltip: 'Diff [3]'
    }],
    selectedIndex: index,
    onSelectIndex: function onSelectIndex(idx) {
      return setIndex(idx);
    }
  });
};

exports.OutlineThreeButtonsSmall = OutlineThreeButtonsSmall;
OutlineThreeButtonsSmall.story = {
  name: 'outline three buttons small'
};

var PillWText = function PillWText() {
  return /*#__PURE__*/_react["default"].createElement(_ButtonToggle.ButtonToggle, {
    titles: [{
      title: '360',
      tooltip: 'View at 360px'
    }, {
      title: '764',
      tooltip: 'View at 764px'
    }, {
      title: '1024',
      tooltip: 'View at 1024px'
    }],
    selectedIndex: 1,
    onSelectIndex: onSelectIndex,
    appearance: "pill"
  });
};

exports.PillWText = PillWText;
PillWText.story = {
  name: 'pill w/text'
};

var PillWImage = function PillWImage() {
  return /*#__PURE__*/_react["default"].createElement(_ButtonToggle.ButtonToggle, {
    titles: [{
      title: /*#__PURE__*/_react["default"].createElement("img", {
        src: _chrome["default"],
        alt: "Chrome"
      }),
      tooltip: 'View Chrome'
    }, {
      title: /*#__PURE__*/_react["default"].createElement("img", {
        src: _firefox["default"],
        alt: "Firefox"
      }),
      tooltip: 'View Firefox'
    }, {
      title: /*#__PURE__*/_react["default"].createElement("img", {
        src: _ie["default"],
        alt: "Safari"
      }),
      tooltip: 'View Safari'
    }, {
      title: /*#__PURE__*/_react["default"].createElement("img", {
        src: _safari["default"],
        alt: "IE"
      }),
      tooltip: 'View IE'
    }],
    selectedIndex: 1,
    onSelectIndex: onSelectIndex,
    appearance: "pill"
  });
};

exports.PillWImage = PillWImage;
PillWImage.story = {
  name: 'pill w/image'
};

var Tab = function Tab() {
  return /*#__PURE__*/_react["default"].createElement(_ButtonToggle.ButtonToggle, {
    titles: [{
      title: 'Canvas',
      tooltip: 'View your story'
    }, {
      title: 'Snapshot',
      tooltip: 'View a screenshot of your story'
    }, {
      title: 'Docs',
      tooltip: 'View docs for this story'
    }],
    selectedIndex: 1,
    onSelectIndex: onSelectIndex,
    appearance: "tab"
  });
};

exports.Tab = Tab;
Tab.story = {
  name: 'tab'
};
