// @storybook/design-system/dist/components/Cardinal.stories.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.LargePlural = exports.LargeSingular = exports.LargeLink = exports.LargeDefault = exports.LargeLoading = exports.SmallSelectable = exports.SmallNeutral = exports.SmallWarning = exports.SmallNegative = exports.SmallPositive = exports.SmallDefault = exports.SmallLoading = exports.Alignment = exports.AllCardinals = exports["default"] = void 0;

var _addonActions = require("@storybook/addon-actions");

var _react = _interopRequireDefault(require("react"));

var _Cardinal = require("./Cardinal");

var onHover = (0, _addonActions.action)('hover');
var onClick = (0, _addonActions.action)('click');
var _default = {
  title: 'Cardinal'
};
exports["default"] = _default;

var AllCardinals = function AllCardinals() {
  return /*#__PURE__*/_react["default"].createElement("div", {
    style: {
      background: 'white'
    }
  }, /*#__PURE__*/_react["default"].createElement(_Cardinal.Cardinal, {
    isLoading: true,
    size: "small"
  }), /*#__PURE__*/_react["default"].createElement(_Cardinal.Cardinal, {
    size: "small",
    count: 14,
    text: "Changes"
  }), /*#__PURE__*/_react["default"].createElement(_Cardinal.Cardinal, {
    size: "small",
    count: 7,
    text: "Accepted",
    noPlural: true,
    status: "positive"
  }), /*#__PURE__*/_react["default"].createElement(_Cardinal.Cardinal, {
    size: "small",
    count: 4,
    text: "Denied",
    noPlural: true,
    status: "negative"
  }), /*#__PURE__*/_react["default"].createElement(_Cardinal.Cardinal, {
    size: "small",
    count: 3,
    text: "Unreviewed",
    noPlural: true,
    status: "warning"
  }), /*#__PURE__*/_react["default"].createElement(_Cardinal.Cardinal, {
    size: "small",
    count: 0,
    text: "Unreviewed",
    noPlural: true,
    status: "neutral"
  }), /*#__PURE__*/_react["default"].createElement("br", null), /*#__PURE__*/_react["default"].createElement(_Cardinal.Cardinal, {
    selectable: true,
    onHover: onHover,
    onClick: onClick,
    active: true,
    size: "small",
    count: 14,
    text: "Selectable",
    noPlural: true
  }), /*#__PURE__*/_react["default"].createElement(_Cardinal.Cardinal, {
    selectable: true,
    onHover: onHover,
    onClick: onClick,
    size: "small",
    count: 14,
    text: "Selectable",
    noPlural: true
  }), /*#__PURE__*/_react["default"].createElement(_Cardinal.Cardinal, {
    selectable: true,
    onHover: onHover,
    onClick: onClick,
    size: "small",
    count: 14,
    text: "Selectable",
    noPlural: true
  }), /*#__PURE__*/_react["default"].createElement("br", null), /*#__PURE__*/_react["default"].createElement(_Cardinal.Cardinal, {
    isLoading: true
  }), /*#__PURE__*/_react["default"].createElement(_Cardinal.Cardinal, {
    count: 37,
    text: "Stories"
  }), /*#__PURE__*/_react["default"].createElement(_Cardinal.Cardinal, {
    count: 12,
    countLink: "http://github.com",
    text: "Commits"
  }), /*#__PURE__*/_react["default"].createElement(_Cardinal.Cardinal, {
    size: "large",
    count: 1,
    text: "Story"
  }), /*#__PURE__*/_react["default"].createElement(_Cardinal.Cardinal, {
    size: "large",
    count: 2,
    text: "Story"
  }));
};

exports.AllCardinals = AllCardinals;
AllCardinals.story = {
  name: 'all cardinals'
};

var Alignment = function Alignment() {
  return /*#__PURE__*/_react["default"].createElement("div", {
    style: {
      background: 'white'
    }
  }, /*#__PURE__*/_react["default"].createElement(_Cardinal.Cardinal, {
    size: "small",
    count: 14,
    text: "Changes",
    alignment: "left"
  }), /*#__PURE__*/_react["default"].createElement(_Cardinal.Cardinal, {
    size: "small",
    count: 14,
    text: "Changes",
    alignment: "center"
  }), /*#__PURE__*/_react["default"].createElement(_Cardinal.Cardinal, {
    size: "small",
    count: 14,
    text: "Changes",
    alignment: "right"
  }), /*#__PURE__*/_react["default"].createElement("br", null), /*#__PURE__*/_react["default"].createElement(_Cardinal.Cardinal, {
    size: "large",
    count: 2,
    text: "Story",
    alignment: "left"
  }), /*#__PURE__*/_react["default"].createElement(_Cardinal.Cardinal, {
    size: "large",
    count: 2,
    text: "Story",
    alignment: "center"
  }), /*#__PURE__*/_react["default"].createElement(_Cardinal.Cardinal, {
    size: "large",
    count: 2,
    text: "Story",
    alignment: "right"
  }));
};

exports.Alignment = Alignment;
Alignment.story = {
  name: 'control alignment'
};

var SmallLoading = function SmallLoading() {
  return /*#__PURE__*/_react["default"].createElement(_Cardinal.Cardinal, {
    isLoading: true,
    size: "small"
  });
};

exports.SmallLoading = SmallLoading;
SmallLoading.story = {
  name: 'small loading'
};

var SmallDefault = function SmallDefault() {
  return /*#__PURE__*/_react["default"].createElement(_Cardinal.Cardinal, {
    size: "small",
    count: 14,
    text: "Changes"
  });
};

exports.SmallDefault = SmallDefault;
SmallDefault.story = {
  name: 'small default'
};

var SmallPositive = function SmallPositive() {
  return /*#__PURE__*/_react["default"].createElement(_Cardinal.Cardinal, {
    size: "small",
    count: 7,
    text: "Accepted",
    noPlural: true,
    status: "positive"
  });
};

exports.SmallPositive = SmallPositive;
SmallPositive.story = {
  name: 'small positive'
};

var SmallNegative = function SmallNegative() {
  return /*#__PURE__*/_react["default"].createElement(_Cardinal.Cardinal, {
    size: "small",
    count: 4,
    text: "Denied",
    noPlural: true,
    status: "negative"
  });
};

exports.SmallNegative = SmallNegative;
SmallNegative.story = {
  name: 'small negative'
};

var SmallWarning = function SmallWarning() {
  return /*#__PURE__*/_react["default"].createElement(_Cardinal.Cardinal, {
    size: "small",
    count: 3,
    text: "Unreviewed",
    noPlural: true,
    status: "warning"
  });
};

exports.SmallWarning = SmallWarning;
SmallWarning.story = {
  name: 'small warning'
};

var SmallNeutral = function SmallNeutral() {
  return /*#__PURE__*/_react["default"].createElement(_Cardinal.Cardinal, {
    size: "small",
    count: 0,
    text: "Unreviewed",
    noPlural: true,
    status: "neutral"
  });
};

exports.SmallNeutral = SmallNeutral;
SmallNeutral.story = {
  name: 'small neutral'
};

var SmallSelectable = function SmallSelectable() {
  return /*#__PURE__*/_react["default"].createElement("div", null, /*#__PURE__*/_react["default"].createElement(_Cardinal.Cardinal, {
    selectable: true,
    onHover: onHover,
    onClick: onClick,
    active: true,
    size: "small",
    count: 14,
    text: "Selectable",
    noPlural: true
  }), /*#__PURE__*/_react["default"].createElement(_Cardinal.Cardinal, {
    selectable: true,
    onHover: onHover,
    onClick: onClick,
    size: "small",
    count: 14,
    text: "Selectable",
    noPlural: true
  }));
};

exports.SmallSelectable = SmallSelectable;
SmallSelectable.story = {
  name: 'small selectable'
};

var LargeLoading = function LargeLoading() {
  return /*#__PURE__*/_react["default"].createElement(_Cardinal.Cardinal, {
    isLoading: true,
    size: "large"
  });
};

exports.LargeLoading = LargeLoading;
LargeLoading.story = {
  name: 'large loading'
};

var LargeDefault = function LargeDefault() {
  return /*#__PURE__*/_react["default"].createElement(_Cardinal.Cardinal, {
    size: "large",
    count: 37,
    text: "Stories"
  });
};

exports.LargeDefault = LargeDefault;
LargeDefault.story = {
  name: 'large default'
};

var LargeLink = function LargeLink() {
  return /*#__PURE__*/_react["default"].createElement(_Cardinal.Cardinal, {
    size: "large",
    count: 12,
    countLink: "http://github.com",
    text: "Commits"
  });
};

exports.LargeLink = LargeLink;
LargeLink.story = {
  name: 'large link'
};

var LargeSingular = function LargeSingular() {
  return /*#__PURE__*/_react["default"].createElement(_Cardinal.Cardinal, {
    size: "large",
    count: 1,
    text: "Story"
  });
};

exports.LargeSingular = LargeSingular;
LargeSingular.story = {
  name: 'large singular'
};

var LargePlural = function LargePlural() {
  return /*#__PURE__*/_react["default"].createElement(_Cardinal.Cardinal, {
    size: "large",
    count: 2,
    text: "Story"
  });
};

exports.LargePlural = LargePlural;
LargePlural.story = {
  name: 'large plural'
};
