// @storybook/design-system/dist/components/Cardinal.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Cardinal = Cardinal;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _taggedTemplateLiteral2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteral"));

var _pluralize = _interopRequireDefault(require("pluralize"));

var _polished = require("polished");

var _react = _interopRequireDefault(require("react"));

var _styledComponents = _interopRequireWildcard(require("styled-components"));

var _Link = require("./Link");

var _styles = require("./shared/styles");

var _animation = require("./shared/animation");

function _templateObject11() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n      background: ", ";\n    "]);

  _templateObject11 = function _templateObject11() {
    return data;
  };

  return data;
}

function _templateObject10() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n      cursor: pointer;\n\n      &:hover {\n        background: ", ";\n      }\n    "]);

  _templateObject10 = function _templateObject10() {
    return data;
  };

  return data;
}

function _templateObject9() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n      ", ", ", " {\n        overflow: hidden;\n\n        > span,\n        a {\n          ", ";\n        }\n\n        img {\n          opacity: 0;\n        }\n      }\n    "]);

  _templateObject9 = function _templateObject9() {
    return data;
  };

  return data;
}

function _templateObject8() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  display: inline-block;\n  vertical-align: top;\n  padding: 8px 12px;\n  border-radius: ", "px;\n  text-align: ", ";\n\n  svg {\n    height: 12px;\n    margin: 1px 0;\n    vertical-align: top;\n    color: inherit;\n  }\n\n  ", ";\n\n  ", ";\n\n  ", ";\n\n  ", " {\n    font-weight: ", ";\n    font-size: ", "px;\n    line-height: ", "px;\n    margin-bottom: ", "px;\n  }\n\n  ", " {\n    color: ", ";\n    font-size: ", "px;\n    line-height: ", "px;\n    clear: both;\n  }\n"]);

  _templateObject8 = function _templateObject8() {
    return data;
  };

  return data;
}

function _templateObject7() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  span {\n    display: inline-block;\n  }\n"]);

  _templateObject7 = function _templateObject7() {
    return data;
  };

  return data;
}

function _templateObject6() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n      color: ", ";\n      text-decoration: none;\n      &:hover {\n        color: ", ";\n      }\n      &:active {\n        color: ", ";\n      }\n    "]);

  _templateObject6 = function _templateObject6() {
    return data;
  };

  return data;
}

function _templateObject5() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n      color: ", ";\n    "]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n      color: ", ";\n    "]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n      color: ", ";\n    "]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n      color: ", ";\n    "]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  color: ", ";\n  display: block;\n\n  ", ";\n  ", ";\n  ", ";\n  ", ";\n  ", ";\n\n  span {\n    display: inline-block;\n  }\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

var Count = _styledComponents["default"].div(_templateObject(), _styles.color.darker, function (props) {
  return props.status === 'positive' && (0, _styledComponents.css)(_templateObject2(), _styles.color.positive);
}, function (props) {
  return props.status === 'negative' && (0, _styledComponents.css)(_templateObject3(), _styles.color.negative);
}, function (props) {
  return props.status === 'warning' && (0, _styledComponents.css)(_templateObject4(), _styles.color.warning);
}, function (props) {
  return props.status === 'neutral' && (0, _styledComponents.css)(_templateObject5(), _styles.color.dark);
}, function (props) {
  return props.status === 'link' && (0, _styledComponents.css)(_templateObject6(), _styles.color.secondary, (0, _polished.darken)(0.07, _styles.color.secondary), (0, _polished.darken)(0.1, _styles.color.secondary));
});

var Text = _styledComponents["default"].div(_templateObject7());

var CardinalInner = _styledComponents["default"].div(_templateObject8(), _styles.spacing.borderRadius.small, function (props) {
  return props.alignment;
}, function (props) {
  return props.isLoading && (0, _styledComponents.css)(_templateObject9(), Count, Text, _animation.inlineGlow);
}, function (props) {
  return props.selectable && (0, _styledComponents.css)(_templateObject10(), _styles.background.app);
}, function (props) {
  return props.active && (0, _styledComponents.css)(_templateObject11(), _styles.background.app);
}, Count, function (props) {
  return props.size === 'small' ? _styles.typography.weight.bold : _styles.typography.weight.regular;
}, function (props) {
  return props.size === 'small' ? _styles.typography.size.s2 : _styles.typography.size.m3;
}, function (props) {
  return props.size === 'small' ? _styles.typography.size.s3 : _styles.typography.size.m3;
}, function (props) {
  return props.size === 'small' ? 2 : 4;
}, Text, _styles.color.dark, function (props) {
  return props.size === 'small' ? _styles.typography.size.s1 : _styles.typography.size.s2;
}, function (props) {
  return props.size === 'small' ? _styles.typography.size.s2 : _styles.typography.size.m1;
});

function Cardinal(_ref) {
  var isLoading = _ref.isLoading,
      selectable = _ref.selectable,
      onHover = _ref.onHover,
      onClick = _ref.onClick,
      active = _ref.active,
      size = _ref.size,
      count = _ref.count,
      countLink = _ref.countLink,
      text = _ref.text,
      noPlural = _ref.noPlural,
      status = _ref.status,
      CountWrapper = _ref.CountWrapper,
      TextWrapper = _ref.TextWrapper,
      alignment = _ref.alignment,
      props = (0, _objectWithoutProperties2["default"])(_ref, ["isLoading", "selectable", "onHover", "onClick", "active", "size", "count", "countLink", "text", "noPlural", "status", "CountWrapper", "TextWrapper", "alignment"]);
  var countValue = count;

  if (countLink) {
    countValue = /*#__PURE__*/_react["default"].createElement(_Link.Link, {
      href: countLink
    }, count);
  }

  var events = {};

  if (selectable) {
    events = {
      onMouseEnter: function onMouseEnter() {
        return onHover(true);
      },
      onMouseLeave: function onMouseLeave() {
        return onHover(false);
      },
      onClick: onClick
    };
  }

  return /*#__PURE__*/_react["default"].createElement(CardinalInner, (0, _extends2["default"])({
    isLoading: isLoading,
    selectable: selectable,
    active: active,
    size: size,
    alignment: alignment
  }, events, props), /*#__PURE__*/_react["default"].createElement(Count, {
    status: status
  }, /*#__PURE__*/_react["default"].createElement(CountWrapper, null, countValue)), /*#__PURE__*/_react["default"].createElement(Text, null, /*#__PURE__*/_react["default"].createElement(TextWrapper, null, !noPlural && typeof count === 'number' ? (0, _pluralize["default"])(text, count) : text)));
}

var DefaultWrapper = function DefaultWrapper(_ref2) {
  var children = _ref2.children;
  return /*#__PURE__*/_react["default"].createElement("span", null, children);
};

Cardinal.defaultProps = {
  isLoading: false,
  selectable: false,
  onHover: function onHover() {
    return 0;
  },
  onClick: function onClick() {
    return 0;
  },
  active: false,
  size: 'large',
  status: 'default',
  count: '000',
  countLink: null,
  noPlural: false,
  text: 'loading',
  CountWrapper: DefaultWrapper,
  TextWrapper: DefaultWrapper,
  alignment: 'left'
};
