// @storybook/design-system/dist/components/Checkbox.stories.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Checked = exports.Unchecked = exports.All = exports.Template = exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _addonActions = require("@storybook/addon-actions");

var _Checkbox = require("./Checkbox");

var onChange = (0, _addonActions.action)('change');
var _default = {
  title: 'forms/Checkbox',
  component: _Checkbox.Checkbox
};
exports["default"] = _default;

var Template = function Template(args) {
  return /*#__PURE__*/_react["default"].createElement(_Checkbox.Checkbox, args);
};

exports.Template = Template;
Template.args = {
  label: 'Basic',
  hideLabel: false
};
Template.story = {
  name: 'Playground'
};

var All = function All() {
  return /*#__PURE__*/_react["default"].createElement("form", null, /*#__PURE__*/_react["default"].createElement(_Checkbox.Checkbox, {
    id: "Unchecked",
    label: "Cats",
    hideLabel: true,
    onChange: onChange
  }), /*#__PURE__*/_react["default"].createElement(_Checkbox.Checkbox, {
    id: "Checked",
    label: "Cats",
    hideLabel: true,
    checked: true,
    onChange: onChange
  }), /*#__PURE__*/_react["default"].createElement(_Checkbox.Checkbox, {
    id: "With-label-and-error",
    label: "Cats",
    onChange: onChange,
    error: "There's a snake in my boots"
  }), /*#__PURE__*/_react["default"].createElement(_Checkbox.Checkbox, {
    id: "With-label",
    label: "Cats",
    onChange: onChange
  }), /*#__PURE__*/_react["default"].createElement(_Checkbox.Checkbox, {
    appearance: "secondary",
    id: "With-label",
    label: "Secondary",
    checked: true,
    onChange: onChange
  }), /*#__PURE__*/_react["default"].createElement(_Checkbox.Checkbox, {
    appearance: "secondary",
    id: "With-label",
    label: "Secondary",
    onChange: onChange
  }));
};

exports.All = All;
var Unchecked = Template.bind();
exports.Unchecked = Unchecked;
Unchecked.args = {
  id: 'Unchecked',
  label: 'Cats',
  hideLabel: true
};
var Checked = Template.bind();
exports.Checked = Checked;
Checked.args = {
  id: 'Checked',
  label: 'Cats',
  hideLabel: true,
  checked: true
};
