// @storybook/design-system/dist/components/FormErrorState.stories.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.PureMultipleErrors = exports.Default = exports["default"] = void 0;

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var _taggedTemplateLiteral2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteral"));

var _react = _interopRequireWildcard(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _addonActions = require("@storybook/addon-actions");

var _FormErrorState = require("./FormErrorState");

var _Button = require("./Button");

var _Input = require("./Input");

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { (0, _defineProperty2["default"])(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

function _templateObject2() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  padding-bottom: 1em;\n"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  padding: 3em 12em;\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

var _default = {
  title: 'forms/FormErrorState',
  component: _FormErrorState.FormErrorState
};
exports["default"] = _default;

var FormWrapper = _styledComponents["default"].div(_templateObject());

var Input = (0, _styledComponents["default"])(_Input.Input)(_templateObject2());

var onSubmit = function onSubmit(e) {
  e.preventDefault();
  (0, _addonActions.action)('onClick')(e);
};

var Children = function Children(_ref) {
  var handleSubmit = _ref.onSubmit,
      getFormErrorFieldProps = _ref.getFormErrorFieldProps;

  // Use whatever form state management you need. This is just an example.
  var _useState = (0, _react.useState)(''),
      _useState2 = (0, _slicedToArray2["default"])(_useState, 2),
      input1Value = _useState2[0],
      setInput1Value = _useState2[1];

  var _useState3 = (0, _react.useState)(''),
      _useState4 = (0, _slicedToArray2["default"])(_useState3, 2),
      input2Value = _useState4[0],
      setInput2Value = _useState4[1];

  return (
    /*#__PURE__*/
    // @ts-ignore
    _react["default"].createElement("form", {
      onSubmit: handleSubmit
    }, /*#__PURE__*/_react["default"].createElement(Input, (0, _extends2["default"])({
      id: "input-1"
    }, getFormErrorFieldProps({
      id: 'input-1',
      validate: function validate(value) {
        return !value && "There is an error with this field with value: \"".concat(value, "\"");
      }
    }), {
      value: input1Value,
      onChange: function onChange(e) {
        (0, _addonActions.action)('change')(e.target.value);
        setInput1Value(e.target.value);
      },
      startFocused: true,
      appearance: "secondary",
      label: "Label",
      hideLabel: true
    })), /*#__PURE__*/_react["default"].createElement(Input, (0, _extends2["default"])({
      id: "input-2"
    }, getFormErrorFieldProps({
      id: 'input-2',
      validate: function validate(value) {
        return "There is an error with this field with value: \"".concat(value, "\"");
      }
    }), {
      value: input2Value,
      onChange: function onChange(e) {
        (0, _addonActions.action)('change')(e.target.value);
        setInput2Value(e.target.value);
      },
      appearance: "secondary",
      label: "Label",
      hideLabel: true
    })), /*#__PURE__*/_react["default"].createElement(_Button.Button, {
      appearance: "secondary",
      type: "submit"
    }, "Submit"))
  );
};

var decorators = [function (storyFn) {
  return /*#__PURE__*/_react["default"].createElement(FormWrapper, null, storyFn());
}];

var Default = function Default(args) {
  return /*#__PURE__*/_react["default"].createElement(_FormErrorState.FormErrorState, args);
};

exports.Default = Default;
Default.decorators = decorators;
Default.args = {
  onSubmit: onSubmit,
  children: Children
};
var pureActions = (0, _addonActions.actions)('onFocus', 'onBlur', 'onMouseEnter', 'onMouseLeave', 'trackErrors');

var PureMultipleErrors = function PureMultipleErrors(args) {
  return /*#__PURE__*/_react["default"].createElement(_FormErrorState.PureFormErrorState, args);
};

exports.PureMultipleErrors = PureMultipleErrors;
PureMultipleErrors.decorators = decorators;
PureMultipleErrors.args = _objectSpread(_objectSpread({}, pureActions), {}, {
  getError: function getError(args) {
    return "There is an error with this field with value: \"".concat(args.value, "\"");
  },
  primaryFieldId: 'input-2',
  blurredFieldIds: new Set(['input-1', 'input-2']),
  children: Children
});
