// @storybook/design-system/dist/components/Highlight.stories.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.WithComponentChildren = exports.CustomStyling = exports.Wrapping = exports.Markdown = exports.YAML = exports.JSON = exports.CSS = exports.TSX = exports.Typescript = exports.JSX = exports.Javascript = exports.Bash = exports.Basic = exports["default"] = exports.typescriptCodeWithWrappers = exports.javascriptCodeWithWrappers = void 0;

var _taggedTemplateLiteral2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteral"));

var _react = _interopRequireDefault(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _Highlight = require("./Highlight");

function _templateObject2() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  code,\n  pre {\n    font-size: 16px;\n  }\n"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  max-width: 300px;\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

var bashCode = "# Highlight bash:\nnpx install some-package-name\ncd some-package-name";
var bashCodeWithWrappers = "<pre class=\"language-bash\"><code class=\"language-bash\">".concat(bashCode, "</code></pre>");
var javascriptCode = "// Highlight JavaScript:\nimport React from 'react'\n\nconst MyComponent = () => (\n  &#x3C;div>My component renders all the things&#x3C;/div>\n)\n\nexport default MyComponent\n";
var javascriptCodeWithWrappers = "<pre class=\"language-javascript\"><code class=\"language-javascript\">".concat(javascriptCode, "</code></pre>");
exports.javascriptCodeWithWrappers = javascriptCodeWithWrappers;
var jsxCode = javascriptCode;
var jsxCodeWithWrappers = "<pre class=\"language-jsx\"><code class=\"language-jsx\">".concat(jsxCode, "</code></pre>");
var typescriptCode = "// Highlight Typescript:\nimport React from 'react'\n\ninterface InterfaceMyComponentProps {\n  isCool: boolean;\n}\n\nconst MyComponent: React.SFC&#x3C;InterfaceMyComponentProps> = ({ isCool }) => (\n  &#x3C;div>My component isCool: {isCool}&#x3C;/div>\n)\n\nexport default MyComponent\n";
var typescriptCodeWithWrappers = "<pre class=\"language-typescript\"><code class=\"language-typescript\">".concat(typescriptCode, "</code></pre>");
exports.typescriptCodeWithWrappers = typescriptCodeWithWrappers;
var tsxCode = typescriptCode;
var tsxCodeWithWrappers = "<pre class=\"language-tsx\"><code class=\"language-tsx\">".concat(tsxCode, "</code></pre>");
var cssCode = "/* Highlight CSS: */\n.someClass {\n  position: relative;\n}\n";
var cssCodeWithWrappers = "<pre class=\"language-css\"><code class=\"language-css\">".concat(cssCode, "</code></pre>");
var jsonCode = "{\n  \"name\": \"@yourorg/package\",\n  \"version\": \"0.0.1\",\n  \"description\": \"A sweet package\",\n  \"repository\": {\n    \"type\": \"git\",\n    \"url\": \"https://github.com/yourorg/package.git\"\n  }\n}";
var jsonCodeWithWrappers = "<pre class=\"language-json\"><code class=\"language-json\">".concat(jsonCode, "</code></pre>");
var yamlCode = "version: 2\njobs:\n  build:\n    docker:\n      - image: circleci/node:8.10.0\n\n    working_directory: ~/repo\n\n    steps:\n      - checkout\n";
var yamlCodeWithWrappers = "<pre class=\"language-yaml\"><code class=\"language-yaml\">".concat(yamlCode, "</code></pre>");
var markdownCode = "import { Meta, Story, Preview } from '@storybook/addon-docs/blocks';\n\n&#x3C;Meta title=\"MDX/Checkbox\" component={Checkbox} />\n\n# This is a level 1 heading\n\n## This is a level 2 heading\n\n### This is a level 3 heading\n\n#### This is a level 4 heading\n\nThis is a paragraph\n\n| Table | Heading |\n|-------|---------|\n| row   | value   |\n\nexport const test = () => &#x3C;Derp prop=\"test\" />\n";
var markdownCodeWithWrappers = "<pre class=\"language-markdown\"><code class=\"language-markdown\">".concat(markdownCode, "</code></pre>");
var _default = {
  title: 'Highlight',
  component: _Highlight.Highlight,
  excludeStories: ['javascriptCodeWithWrappers', 'typescriptCodeWithWrappers']
};
exports["default"] = _default;

var Basic = function Basic(args) {
  return /*#__PURE__*/_react["default"].createElement(_Highlight.Highlight, args);
};

exports.Basic = Basic;
Basic.args = {};

var Bash = function Bash() {
  return /*#__PURE__*/_react["default"].createElement(_react["default"].Fragment, null, /*#__PURE__*/_react["default"].createElement("strong", null, "Autoformat"), /*#__PURE__*/_react["default"].createElement(_Highlight.Highlight, {
    language: "bash"
  }, bashCode), /*#__PURE__*/_react["default"].createElement("strong", null, "Pre-formatted"), /*#__PURE__*/_react["default"].createElement(_Highlight.Highlight, null, bashCodeWithWrappers));
};

exports.Bash = Bash;

var Javascript = function Javascript() {
  return /*#__PURE__*/_react["default"].createElement(_react["default"].Fragment, null, /*#__PURE__*/_react["default"].createElement("strong", null, "Autoformat"), /*#__PURE__*/_react["default"].createElement(_Highlight.Highlight, {
    language: "javascript"
  }, javascriptCode), /*#__PURE__*/_react["default"].createElement("strong", null, "Pre-formatted"), /*#__PURE__*/_react["default"].createElement(_Highlight.Highlight, null, javascriptCodeWithWrappers));
};

exports.Javascript = Javascript;

var JSX = function JSX() {
  return /*#__PURE__*/_react["default"].createElement(_react["default"].Fragment, null, /*#__PURE__*/_react["default"].createElement("strong", null, "Autoformat"), /*#__PURE__*/_react["default"].createElement(_Highlight.Highlight, {
    language: "jsx"
  }, jsxCode), /*#__PURE__*/_react["default"].createElement("strong", null, "Pre-formatted"), /*#__PURE__*/_react["default"].createElement(_Highlight.Highlight, null, jsxCodeWithWrappers));
};

exports.JSX = JSX;

var Typescript = function Typescript() {
  return /*#__PURE__*/_react["default"].createElement(_react["default"].Fragment, null, /*#__PURE__*/_react["default"].createElement("strong", null, "Autoformat"), /*#__PURE__*/_react["default"].createElement(_Highlight.Highlight, {
    language: "typescript"
  }, typescriptCode), /*#__PURE__*/_react["default"].createElement("strong", null, "Pre-formatted"), /*#__PURE__*/_react["default"].createElement(_Highlight.Highlight, null, typescriptCodeWithWrappers));
};

exports.Typescript = Typescript;

var TSX = function TSX() {
  return /*#__PURE__*/_react["default"].createElement(_react["default"].Fragment, null, /*#__PURE__*/_react["default"].createElement("strong", null, "Autoformat"), /*#__PURE__*/_react["default"].createElement(_Highlight.Highlight, {
    language: "tsx"
  }, tsxCode), /*#__PURE__*/_react["default"].createElement("strong", null, "Pre-formatted"), /*#__PURE__*/_react["default"].createElement(_Highlight.Highlight, null, tsxCodeWithWrappers));
};

exports.TSX = TSX;

var CSS = function CSS() {
  return /*#__PURE__*/_react["default"].createElement(_react["default"].Fragment, null, /*#__PURE__*/_react["default"].createElement("strong", null, "Autoformat"), /*#__PURE__*/_react["default"].createElement(_Highlight.Highlight, {
    language: "css"
  }, cssCode), /*#__PURE__*/_react["default"].createElement("strong", null, "Pre-formatted"), /*#__PURE__*/_react["default"].createElement(_Highlight.Highlight, null, cssCodeWithWrappers));
};

exports.CSS = CSS;

var JSON = function JSON() {
  return /*#__PURE__*/_react["default"].createElement(_react["default"].Fragment, null, /*#__PURE__*/_react["default"].createElement("strong", null, "Autoformat"), /*#__PURE__*/_react["default"].createElement(_Highlight.Highlight, {
    language: "json"
  }, jsonCode), /*#__PURE__*/_react["default"].createElement("strong", null, "Pre-formatted"), /*#__PURE__*/_react["default"].createElement(_Highlight.Highlight, null, jsonCodeWithWrappers));
};

exports.JSON = JSON;

var YAML = function YAML() {
  return /*#__PURE__*/_react["default"].createElement(_react["default"].Fragment, null, /*#__PURE__*/_react["default"].createElement("strong", null, "Autoformat"), /*#__PURE__*/_react["default"].createElement(_Highlight.Highlight, {
    language: "yaml"
  }, yamlCode), /*#__PURE__*/_react["default"].createElement("strong", null, "Pre-formatted"), /*#__PURE__*/_react["default"].createElement(_Highlight.Highlight, null, yamlCodeWithWrappers));
};

exports.YAML = YAML;

var Markdown = function Markdown() {
  return /*#__PURE__*/_react["default"].createElement(_react["default"].Fragment, null, /*#__PURE__*/_react["default"].createElement("strong", null, "Autoformat"), /*#__PURE__*/_react["default"].createElement(_Highlight.Highlight, {
    language: "markdown"
  }, markdownCode), /*#__PURE__*/_react["default"].createElement("strong", null, "Autoformat w/ MDX Language"), /*#__PURE__*/_react["default"].createElement(_Highlight.Highlight, {
    language: "mdx"
  }, markdownCode), /*#__PURE__*/_react["default"].createElement("strong", null, "Autoformat w/ MD Language"), /*#__PURE__*/_react["default"].createElement(_Highlight.Highlight, {
    language: "md"
  }, markdownCode), /*#__PURE__*/_react["default"].createElement("strong", null, "Pre-formatted"), /*#__PURE__*/_react["default"].createElement(_Highlight.Highlight, null, markdownCodeWithWrappers));
};

exports.Markdown = Markdown;

var SmallContainer = _styledComponents["default"].div(_templateObject());

var Wrapping = function Wrapping() {
  return /*#__PURE__*/_react["default"].createElement(SmallContainer, null, /*#__PURE__*/_react["default"].createElement(_Highlight.Highlight, {
    language: "javascript"
  }, javascriptCode));
};

exports.Wrapping = Wrapping;
var StyledHighlight = (0, _styledComponents["default"])(_Highlight.Highlight)(_templateObject2());

var CustomStyling = function CustomStyling() {
  return /*#__PURE__*/_react["default"].createElement(StyledHighlight, {
    language: "json"
  }, jsonCode);
};
/* eslint-disable */


exports.CustomStyling = CustomStyling;

function WrapperComponent(_ref) {
  var children = _ref.children;
  return /*#__PURE__*/_react["default"].createElement("div", {
    dangerouslySetInnerHTML: {
      __html: children
    }
  });
}
/* eslint-enable */


var WithComponentChildren = function WithComponentChildren() {
  return /*#__PURE__*/_react["default"].createElement(_Highlight.Highlight, {
    withHTMLChildren: false
  }, /*#__PURE__*/_react["default"].createElement(WrapperComponent, null, javascriptCodeWithWrappers));
};

exports.WithComponentChildren = WithComponentChildren;
