// @storybook/design-system/dist/components/Input.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Input = exports.PureInput = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _taggedTemplateLiteral2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteral"));

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireWildcard(require("styled-components"));

var _styles = require("./shared/styles");

var _animation = require("./shared/animation");

var _Icon = require("./Icon");

var _Link = require("./Link");

var _WithTooltip = _interopRequireWildcard(require("./tooltip/WithTooltip"));

var _TooltipMessage = require("./tooltip/TooltipMessage");

function _templateObject25() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  position: absolute;\n  right: 0;\n  min-width: 45px;\n  top: 50%;\n  transform: translateY(-50%);\n  font-weight: bold;\n  font-size: 11px;\n  z-index: 2;\n"]);

  _templateObject25 = function _templateObject25() {
    return data;
  };

  return data;
}

function _templateObject24() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  width: 170px;\n"]);

  _templateObject24 = function _templateObject24() {
    return data;
  };

  return data;
}

function _templateObject23() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  width: 100%;\n"]);

  _templateObject23 = function _templateObject23() {
    return data;
  };

  return data;
}

function _templateObject22() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n    display: table-row;\n\n    ", ", ", " {\n      display: table-cell;\n    }\n\n    ", " {\n      width: 1px;\n      padding-right: 20px;\n      vertical-align: middle;\n    }\n\n    ", " {\n      width: auto;\n    }\n\n  "]);

  _templateObject22 = function _templateObject22() {
    return data;
  };

  return data;
}

function _templateObject21() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  ", "\n"]);

  _templateObject21 = function _templateObject21() {
    return data;
  };

  return data;
}

function _templateObject20() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n    ", " {\n        box-shadow: ", " 0 0 0 1px inset;\n        &:focus { box-shadow: ", " 0 0 0 1px inset !important;  }\n    }\n\n    svg {\n      animation: ", " 700ms ease-out;\n      path { fill: ", "; }\n    }\n  "]);

  _templateObject20 = function _templateObject20() {
    return data;
  };

  return data;
}

function _templateObject19() {
  var data = (0, _taggedTemplateLiteral2["default"])([" padding-left: 30px; "]);

  _templateObject19 = function _templateObject19() {
    return data;
  };

  return data;
}

function _templateObject18() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n        left: ", ";\n      "]);

  _templateObject18 = function _templateObject18() {
    return data;
  };

  return data;
}

function _templateObject17() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n        left: 10px;\n      "]);

  _templateObject17 = function _templateObject17() {
    return data;
  };

  return data;
}

function _templateObject16() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n        font-size: ", "px;\n      "]);

  _templateObject16 = function _templateObject16() {
    return data;
  };

  return data;
}

function _templateObject15() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n        font-size: ", "px;\n      "]);

  _templateObject15 = function _templateObject15() {
    return data;
  };

  return data;
}

function _templateObject14() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n    > svg {\n      transition: all 150ms ease-out;\n      position: absolute;\n      top: 50%;\n      ", "\n      height: 1em;\n      width: 1em;\n      margin-top: -.525em;\n      z-index: 3;\n      ", "\n\n      background: transparent;\n\n      path {\n        transition: all 150ms ease-out;\n        fill: ", ";\n      }\n    }\n\n    ", ":focus + svg path {\n      fill: ", ";\n    }\n\n    ", " {\n      padding-left: 40px;\n\n      ", ";      \n    }\n  "]);

  _templateObject14 = function _templateObject14() {
    return data;
  };

  return data;
}

function _templateObject13() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n    ", " {\n      padding-right: 52px;\n    }\n  "]);

  _templateObject13 = function _templateObject13() {
    return data;
  };

  return data;
}

function _templateObject12() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n      font-size: ", "px;\n      line-height: 16px;\n      font-family: ", ";\n      border-radius: ", "px;\n      background: ", ";\n      padding: 8px 10px;\n    "]);

  _templateObject12 = function _templateObject12() {
    return data;
  };

  return data;
}

function _templateObject11() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n      font-size: ", "px;\n      line-height: 16px;\n      padding: 6px 12px; //28px tall\n      border-radius: 3em;\n      background: transparent;\n    "]);

  _templateObject11 = function _templateObject11() {
    return data;
  };

  return data;
}

function _templateObject10() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  display: inline-block;\n  position: relative;\n  vertical-align: top;\n  width: 100%;\n\n  ", " {\n    position: relative;\n    ", "\n\n    background: ", ";\n    color: ", ";\n    font-size: ", "px;\n    line-height: 20px;\n    padding: 10px 15px; //40px tall\n    box-shadow: ", " 0 0 0 1px inset;\n    &:focus { box-shadow: ", " 0 0 0 1px inset; }\n\n\n    ", "\n\n    ", "\n  }\n\n  ", "\n\n  ", "\n\n  ", "\n"]);

  _templateObject10 = function _templateObject10() {
    return data;
  };

  return data;
}

function _templateObject9() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n        border-radius: ", "px;\n      "]);

  _templateObject9 = function _templateObject9() {
    return data;
  };

  return data;
}

function _templateObject8() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n        border-bottom-left-radius: ", "px;\n        border-bottom-right-radius: ", "px;\n        border-top-left-radius: 0;\n        border-top-right-radius: 0;\n        margin-top: -1px;\n        ", "\n      "]);

  _templateObject8 = function _templateObject8() {
    return data;
  };

  return data;
}

function _templateObject7() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n        border-radius: 0px;\n        margin-top: -1px;\n        ", "\n      "]);

  _templateObject7 = function _templateObject7() {
    return data;
  };

  return data;
}

function _templateObject6() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n        border-top-left-radius: ", "px;\n        border-top-right-radius: ", "px;\n        border-bottom-left-radius: 0;\n        border-bottom-right-radius: 0;\n        ", "\n      "]);

  _templateObject6 = function _templateObject6() {
    return data;
  };

  return data;
}

function _templateObject5() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n    position: relative;\n    ", "\n\n    &:focus {\n      z-index: 2;\n    }\n  "]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  ::placeholder {\n    color: ", ";\n  }\n  appearance: none;\n  border:none;\n  box-sizing: border-box;\n  display: block;\n  outline: none;\n  width: 100%;\n  margin: 0;\n\n  &[disabled] {\n    cursor: not-allowed;\n    opacity: .5;\n  }\n\n  &:-webkit-autofill { -webkit-box-shadow: 0 0 0 3em ", " inset; }\n"]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n    border: 0px !important;\n    clip: rect(0 0 0 0) !important;\n    -webkit-clip-path: inset(100%) !important;\n    clip-path: inset(100%) !important;\n    height: 1px !important;\n    overflow: hidden !important;\n    padding: 0px !important;\n    position: absolute !important;\n    white-space: nowrap !important;\n    width: 1px !important;\n  "]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  margin-bottom: 8px;\n  \n  ", "\n"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  font-weight: ", ";\n  font-family: ", ";\n  font-size: ", "px;\n  line-height: ", "px;\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

// prettier-ignore
var Label = _styledComponents["default"].label(_templateObject(), function (props) {
  return props.appearance !== 'code' && _styles.typography.weight.bold;
}, function (props) {
  return props.appearance === 'code' && _styles.typography.type.code;
}, function (props) {
  return props.appearance === 'code' ? _styles.typography.size.s1 - 1 : _styles.typography.size.s2;
}, function (props) {
  return props.appearance === 'code' ? 16 : 20;
}); // prettier-ignore


var LabelWrapper = _styledComponents["default"].div(_templateObject2(), function (props) {
  return props.hideLabel && (0, _styledComponents.css)(_templateObject3());
}); // prettier-ignore


var InputEl = _styledComponents["default"].input(_templateObject4(), _styles.color.mediumdark, _styles.color.lightest);

var getStackLevelStyling = function getStackLevelStyling(props) {
  var radius = 4;
  var stackLevelDefinedStyling = (0, _styledComponents.css)(_templateObject5(), props.error && "z-index: 1;");

  switch (props.stackLevel) {
    case 'top':
      return (0, _styledComponents.css)(_templateObject6(), radius, radius, stackLevelDefinedStyling);

    case 'middle':
      return (0, _styledComponents.css)(_templateObject7(), stackLevelDefinedStyling);

    case 'bottom':
      return (0, _styledComponents.css)(_templateObject8(), radius, radius, stackLevelDefinedStyling);

    default:
      return (0, _styledComponents.css)(_templateObject9(), radius);
  }
}; // prettier-ignore


var InputWrapper = _styledComponents["default"].div(_templateObject10(), InputEl, function (props) {
  return getStackLevelStyling(props);
}, _styles.color.lightest, _styles.color.darkest, _styles.typography.size.s2, _styles.color.border, _styles.color.secondary, function (props) {
  return props.appearance === 'pill' && (0, _styledComponents.css)(_templateObject11(), _styles.typography.size.s1);
}, function (props) {
  return props.appearance === 'code' && (0, _styledComponents.css)(_templateObject12(), _styles.typography.size.s1 - 1, _styles.typography.type.code, _styles.spacing.borderRadius.small, _styles.color.lightest);
}, function (props) {
  return props.startingType === 'password' && (0, _styledComponents.css)(_templateObject13(), InputEl);
}, function (props) {
  return props.icon && (0, _styledComponents.css)(_templateObject14(), props.appearance === 'pill' || props.appearance === 'code' ? (0, _styledComponents.css)(_templateObject15(), _styles.typography.size.s1) : (0, _styledComponents.css)(_templateObject16(), _styles.typography.size.s2), props.appearance === 'pill' || props.appearance === 'code' ? (0, _styledComponents.css)(_templateObject17()) : (0, _styledComponents.css)(_templateObject18(), props.appearance === 'tertiary' ? 0 : "15px"), _styles.color.mediumdark, InputEl, _styles.color.darker, InputEl, (props.appearance === 'pill' || props.appearance === 'code') && (0, _styledComponents.css)(_templateObject19()));
}, function (props) {
  return props.error && (0, _styledComponents.css)(_templateObject20(), InputEl, _styles.color.negative, _styles.color.negative, _animation.jiggle, _styles.color.negative);
}); // prettier-ignore


var InputContainer = _styledComponents["default"].div(_templateObject21(), function (props) {
  return props.orientation === 'horizontal' && (0, _styledComponents.css)(_templateObject22(), LabelWrapper, InputWrapper, LabelWrapper, InputWrapper);
});

var ErrorTooltip = (0, _styledComponents["default"])(_WithTooltip["default"])(_templateObject23());
var ErrorTooltipMessage = (0, _styledComponents["default"])(_TooltipMessage.TooltipMessage)(_templateObject24());

var Action = _styledComponents["default"].div(_templateObject25());

var getErrorMessage = function getErrorMessage(_ref) {
  var error = _ref.error,
      value = _ref.value,
      lastErrorValue = _ref.lastErrorValue;
  var errorMessage = typeof error === 'function' ? error(value) : error;

  if (lastErrorValue) {
    if (value !== lastErrorValue) {
      errorMessage = null;
    }
  }

  return errorMessage;
};

var PureInput = /*#__PURE__*/(0, _react.forwardRef)(function (_ref2, ref) {
  var id = _ref2.id,
      value = _ref2.value,
      label = _ref2.label,
      hideLabel = _ref2.hideLabel,
      orientation = _ref2.orientation,
      icon = _ref2.icon,
      error = _ref2.error,
      appearance = _ref2.appearance,
      errorTooltipPlacement = _ref2.errorTooltipPlacement,
      className = _ref2.className,
      lastErrorValue = _ref2.lastErrorValue,
      startingType = _ref2.startingType,
      type = _ref2.type,
      onActionClick = _ref2.onActionClick,
      stackLevel = _ref2.stackLevel,
      suppressErrorMessage = _ref2.suppressErrorMessage,
      props = (0, _objectWithoutProperties2["default"])(_ref2, ["id", "value", "label", "hideLabel", "orientation", "icon", "error", "appearance", "errorTooltipPlacement", "className", "lastErrorValue", "startingType", "type", "onActionClick", "stackLevel", "suppressErrorMessage"]);

  var _useState = (0, _react.useState)(getErrorMessage({
    error: error,
    value: value,
    lastErrorValue: lastErrorValue
  })),
      _useState2 = (0, _slicedToArray2["default"])(_useState, 2),
      errorMessage = _useState2[0],
      setErrorMessage = _useState2[1];

  var errorId = "".concat(id, "-error");
  (0, _react.useEffect)(function () {
    setErrorMessage(getErrorMessage({
      error: error,
      value: value,
      lastErrorValue: lastErrorValue
    }));
  }, [value, error, lastErrorValue]);

  var inputEl = /*#__PURE__*/_react["default"].createElement(InputEl, (0, _extends2["default"])({
    id: id // Pass the ref to the actual input element so it can be controlled
    // externally.
    ,
    ref: ref,
    value: value,
    type: type,
    "aria-describedby": errorId,
    "aria-invalid": !!error
  }, props));

  return /*#__PURE__*/_react["default"].createElement(InputContainer, {
    orientation: orientation,
    className: className
  }, /*#__PURE__*/_react["default"].createElement(LabelWrapper, {
    hideLabel: hideLabel
  }, /*#__PURE__*/_react["default"].createElement(Label, {
    htmlFor: id,
    appearance: appearance
  }, label)), /*#__PURE__*/_react["default"].createElement(InputWrapper, {
    error: errorMessage,
    "data-error": errorMessage,
    icon: icon,
    appearance: appearance,
    stackLevel: stackLevel,
    startingType: startingType
  }, icon && /*#__PURE__*/_react["default"].createElement(_Icon.Icon, {
    icon: icon,
    "aria-hidden": true
  }), /*#__PURE__*/_react["default"].createElement(ErrorTooltip, {
    tabIndex: -1,
    placement: errorTooltipPlacement,
    trigger: "none",
    startOpen: true,
    tagName: "div",
    hasChrome: !!errorMessage && !suppressErrorMessage,
    tooltip: errorMessage && !suppressErrorMessage && /*#__PURE__*/_react["default"].createElement(ErrorTooltipMessage, {
      desc: errorMessage
    })
  }, inputEl), startingType === 'password' && /*#__PURE__*/_react["default"].createElement(Action, null, /*#__PURE__*/_react["default"].createElement(_Link.Link, {
    isButton: true,
    tertiary: true,
    onClick: onActionClick,
    type: "button"
  }, type === 'password' ? 'Show' : 'Hide'))));
});
exports.PureInput = PureInput;
PureInput.propTypes = {
  id: _propTypes["default"].string.isRequired,
  value: _propTypes["default"].string,
  appearance: _propTypes["default"].oneOf(['default', 'pill', 'code']),
  errorTooltipPlacement: _propTypes["default"].oneOf(_WithTooltip.validPlacements),
  stackLevel: _propTypes["default"].oneOf(['top', 'middle', 'bottom']),
  label: _propTypes["default"].string.isRequired,
  hideLabel: _propTypes["default"].bool,
  orientation: _propTypes["default"].oneOf(['vertical', 'horizontal']),
  icon: _propTypes["default"].string,
  error: _propTypes["default"].oneOfType([_propTypes["default"].node, _propTypes["default"].func]),
  suppressErrorMessage: _propTypes["default"].bool,
  className: _propTypes["default"].string,
  lastErrorValue: _propTypes["default"].string,
  startingType: _propTypes["default"].string,
  type: _propTypes["default"].string,
  onActionClick: _propTypes["default"].func
};
PureInput.defaultProps = {
  value: '',
  appearance: 'default',
  errorTooltipPlacement: 'right',
  stackLevel: undefined,
  hideLabel: false,
  orientation: 'vertical',
  icon: null,
  error: null,
  suppressErrorMessage: false,
  className: null,
  lastErrorValue: null,
  startingType: 'text',
  type: 'text',
  onActionClick: null
};
var Input = /*#__PURE__*/(0, _react.forwardRef)(function (_ref3, ref) {
  var startingType = _ref3.type,
      startFocused = _ref3.startFocused,
      rest = (0, _objectWithoutProperties2["default"])(_ref3, ["type", "startFocused"]);

  var _useState3 = (0, _react.useState)(startingType),
      _useState4 = (0, _slicedToArray2["default"])(_useState3, 2),
      type = _useState4[0],
      setType = _useState4[1];

  var togglePasswordType = (0, _react.useCallback)(function (event) {
    // Make sure this does not submit a form
    event.preventDefault();
    event.stopPropagation();

    if (type === 'password') {
      setType('text');
      return;
    }

    setType('password');
  }, [type, setType]); // Outside refs take precedence

  var inputRef = ref || (0, _react.useRef)();
  var didFocusOnStart = (0, _react.useRef)(false);
  (0, _react.useEffect)(function () {
    if (inputRef && inputRef.current && startFocused && !didFocusOnStart.current) {
      inputRef.current.focus();
      didFocusOnStart.current = true;
    }
  }, [inputRef, inputRef.current, didFocusOnStart, didFocusOnStart.current]);
  return /*#__PURE__*/_react["default"].createElement(PureInput, (0, _extends2["default"])({
    ref: inputRef,
    startingType: startingType,
    type: type,
    onActionClick: togglePasswordType
  }, rest));
});
exports.Input = Input;
Input.propTypes = {
  startFocused: _propTypes["default"].bool,
  type: _propTypes["default"].string
};
Input.defaultProps = {
  startFocused: false,
  type: 'text'
};
