// @storybook/design-system/dist/components/Link.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Link = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _taggedTemplateLiteral2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteral"));

var _react = _interopRequireWildcard(require("react"));

var _styledComponents = _interopRequireWildcard(require("styled-components"));

var _polished = require("polished");

var _Icon = require("./Icon");

var _styles = require("./shared/styles");

function _templateObject10() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  background: none;\n  border: none;\n  padding: 0;\n  font: inherit;\n  cursor: pointer;\n"]);

  _templateObject10 = function _templateObject10() {
    return data;
  };

  return data;
}

function _templateObject9() {
  var data = (0, _taggedTemplateLiteral2["default"])([""]);

  _templateObject9 = function _templateObject9() {
    return data;
  };

  return data;
}

function _templateObject8() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n      color: ", ";\n\n      &:hover {\n        color: ", ";\n      }\n\n      &:active {\n        color: ", ";\n      }\n    "]);

  _templateObject8 = function _templateObject8() {
    return data;
  };

  return data;
}

function _templateObject7() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n      color: inherit;\n\n      &:hover,\n      &:active {\n        color: inherit;\n        text-decoration: underline;\n      }\n    "]);

  _templateObject7 = function _templateObject7() {
    return data;
  };

  return data;
}

function _templateObject6() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n      color: ", ";\n\n      &:hover {\n        color: ", ";\n      }\n\n      &:active {\n        color: ", ";\n      }\n    "]);

  _templateObject6 = function _templateObject6() {
    return data;
  };

  return data;
}

function _templateObject5() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n      color: ", ";\n\n      &:hover {\n        color: ", ";\n      }\n\n      &:active {\n        color: ", ";\n      }\n    "]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n      svg {\n        height: 1em;\n        width: 1em;\n        vertical-align: middle;\n        position: relative;\n        bottom: 0;\n        margin-right: 0;\n      }\n    "]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  display: inline-block;\n  transition: transform 150ms ease-out, color 150ms ease-out;\n  text-decoration: none;\n  color: ", ";\n\n  &:hover,\n  &:focus {\n    cursor: pointer;\n    transform: translateY(-1px);\n    color: ", ";\n  }\n  &:active {\n    transform: translateY(0);\n    color: ", ";\n  }\n\n  svg {\n    display: inline-block;\n    height: 1em;\n    width: 1em;\n    vertical-align: text-top;\n    position: relative;\n    bottom: -0.125em;\n    margin-right: 0.4em;\n  }\n\n  ", ";\n\n  ", ";\n\n  ", ";\n\n  ", ";\n\n  ", ";\n"]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n      > svg:last-of-type {\n        height: 0.7em;\n        width: 0.7em;\n        margin-right: 0;\n        margin-left: 0.25em;\n        bottom: auto;\n        vertical-align: inherit;\n      }\n    "]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  ", ";\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

var LinkInner = _styledComponents["default"].span(_templateObject(), function (props) {
  return props.withArrow && (0, _styledComponents.css)(_templateObject2());
});

var StyledLink = _styledComponents["default"].a(_templateObject3(), _styles.color.secondary, (0, _polished.darken)(0.07, _styles.color.secondary), (0, _polished.darken)(0.1, _styles.color.secondary), function (props) {
  return props.containsIcon && (0, _styledComponents.css)(_templateObject4());
}, function (props) {
  return props.secondary && (0, _styledComponents.css)(_templateObject5(), _styles.color.mediumdark, _styles.color.dark, _styles.color.darker);
}, function (props) {
  return props.tertiary && (0, _styledComponents.css)(_templateObject6(), _styles.color.dark, _styles.color.darkest, _styles.color.mediumdark);
}, function (props) {
  return props.nochrome && (0, _styledComponents.css)(_templateObject7());
}, function (props) {
  return props.inverse && (0, _styledComponents.css)(_templateObject8(), _styles.color.lightest, _styles.color.lighter, _styles.color.light);
});

var UnstyledLink = _styledComponents["default"].a(_templateObject9());

var LinkButton = _styledComponents["default"].button(_templateObject10());
/**
 * Links can contains text and/or icons. Be careful using only icons, you must provide a text alternative via aria-label for accessibility.
 */


// The main purpose of this component is to strip certain props that get passed
// down to the styled component, so that we don't end up passing them to a
// tag which would throw warnings for non-standard props.
var LinkComponentPicker = /*#__PURE__*/(0, _react.forwardRef)(function (_ref, ref) {
  var containsIcon = _ref.containsIcon,
      inverse = _ref.inverse,
      isButton = _ref.isButton,
      LinkWrapper = _ref.LinkWrapper,
      nochrome = _ref.nochrome,
      secondary = _ref.secondary,
      tertiary = _ref.tertiary,
      rest = (0, _objectWithoutProperties2["default"])(_ref, ["containsIcon", "inverse", "isButton", "LinkWrapper", "nochrome", "secondary", "tertiary"]);

  // Use the UnstyledLink here to avoid duplicating styles and creating
  // specificity conflicts by first rendering the StyledLink higher up the chain
  // and then re-rendering it through the 'as' prop.

  /* eslint no-else-return: ["error", { allowElseIf: true }] */
  if (isButton) {
    return /*#__PURE__*/_react["default"].createElement(LinkButton, (0, _extends2["default"])({}, rest, {
      ref: ref
    }));
  } else if (LinkWrapper) {
    return /*#__PURE__*/_react["default"].createElement(LinkWrapper, (0, _extends2["default"])({}, rest, {
      ref: ref
    }));
  }

  return /*#__PURE__*/_react["default"].createElement(UnstyledLink, (0, _extends2["default"])({}, rest, {
    ref: ref
  }));
});
var Link = /*#__PURE__*/(0, _react.forwardRef)(function (_ref2, ref) {
  var children = _ref2.children,
      withArrow = _ref2.withArrow,
      rest = (0, _objectWithoutProperties2["default"])(_ref2, ["children", "withArrow"]);

  var content = /*#__PURE__*/_react["default"].createElement(_react["default"].Fragment, null, /*#__PURE__*/_react["default"].createElement(LinkInner, {
    withArrow: withArrow
  }, children, withArrow && /*#__PURE__*/_react["default"].createElement(_Icon.Icon, {
    icon: "arrowright"
  })));

  return /*#__PURE__*/_react["default"].createElement(StyledLink, (0, _extends2["default"])({
    as: LinkComponentPicker,
    ref: ref
  }, rest), content);
});
exports.Link = Link;
Link.defaultProps = {
  withArrow: false,
  isButton: false,
  containsIcon: false,
  secondary: false,
  tertiary: false,
  nochrome: false,
  inverse: false
};
