// @storybook/design-system/dist/components/Link.stories.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.HasLinkWrapper = exports.IsButton = exports.WithIcon = exports.ContainsIcon = exports.WithArrow = exports.All = exports.Basic = exports["default"] = void 0;

var _taggedTemplateLiteral2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteral"));

var _react = _interopRequireDefault(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _addonActions = require("@storybook/addon-actions");

var _Icon = require("./Icon");

var _Link = require("./Link");

var _StoryLinkWrapper = require("./StoryLinkWrapper");

function _templateObject() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  && {\n    color: red;\n  }\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

var CustomLink = (0, _styledComponents["default"])(_Link.Link)(_templateObject());
var onLinkClick = (0, _addonActions.action)('onLinkClick');
var _default = {
  title: 'Link',
  component: _Link.Link
};
exports["default"] = _default;

var Basic = function Basic() {
  return /*#__PURE__*/_react["default"].createElement(_Link.Link, {
    href: "http://chromatic.com/",
    inverse: false
  }, "link text");
};

exports.Basic = Basic;

var All = function All() {
  return /*#__PURE__*/_react["default"].createElement(_react["default"].Fragment, null, /*#__PURE__*/_react["default"].createElement(_Link.Link, {
    href: "https://learnstorybook.com"
  }, "default"), ' ', /*#__PURE__*/_react["default"].createElement(_Link.Link, {
    secondary: true,
    href: "https://learnstorybook.com"
  }, "secondary"), ' ', /*#__PURE__*/_react["default"].createElement(_Link.Link, {
    tertiary: true,
    href: "https://learnstorybook.com"
  }, "tertiary"), ' ', /*#__PURE__*/_react["default"].createElement(_Link.Link, {
    nochrome: true,
    href: "https://learnstorybook.com"
  }, "nochrome"), ' ', /*#__PURE__*/_react["default"].createElement("span", {
    style: {
      background: '#333'
    }
  }, /*#__PURE__*/_react["default"].createElement(_Link.Link, {
    inverse: true,
    href: "https://learnstorybook.com"
  }, "inverse")));
};

exports.All = All;

var WithArrow = function WithArrow() {
  return /*#__PURE__*/_react["default"].createElement(_Link.Link, {
    containsIcon: true,
    withArrow: true,
    href: "https://learnstorybook.com"
  }, "withArrow shows an arrow behind the link");
};

exports.WithArrow = WithArrow;

var ContainsIcon = function ContainsIcon() {
  return /*#__PURE__*/_react["default"].createElement(_Link.Link, {
    containsIcon: true,
    href: "https://learnstorybook.com",
    "aria-label": "Toggle side bar"
  }, /*#__PURE__*/_react["default"].createElement(_Icon.Icon, {
    icon: "sidebar",
    "aria-hidden": true
  }));
};

exports.ContainsIcon = ContainsIcon;

var WithIcon = function WithIcon() {
  return /*#__PURE__*/_react["default"].createElement(_Link.Link, {
    href: "https://learnstorybook.com"
  }, /*#__PURE__*/_react["default"].createElement(_Icon.Icon, {
    icon: "discord",
    "aria-hidden": true
  }), "Link with an icon in front");
};

exports.WithIcon = WithIcon;

var IsButton = function IsButton() {
  return (
    /*#__PURE__*/

    /* eslint-disable-next-line */
    _react["default"].createElement(_Link.Link, {
      isButton: true,
      onClick: onLinkClick
    }, "is actually a button")
  );
};

exports.IsButton = IsButton;

var HasLinkWrapper = function HasLinkWrapper() {
  return /*#__PURE__*/_react["default"].createElement(_react["default"].Fragment, null, /*#__PURE__*/_react["default"].createElement(_Link.Link, {
    tertiary: true,
    LinkWrapper: _StoryLinkWrapper.StoryLinkWrapper,
    href: "http://storybook.js.org"
  }, "has a LinkWrapper like GatsbyLink or NextLink"), /*#__PURE__*/_react["default"].createElement("br", null), /*#__PURE__*/_react["default"].createElement(CustomLink, {
    tertiary: true,
    LinkWrapper: _StoryLinkWrapper.StoryLinkWrapper,
    href: "http://storybook.js.org"
  }, "has a LinkWrapper like GatsbyLink or NextLink with custom styling"));
};

exports.HasLinkWrapper = HasLinkWrapper;
