// @storybook/design-system/dist/components/ProgressDots.stories.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.LargeComplete = exports.Complete = exports.Halfway = exports.Starting = exports.Loading = exports.Basic = exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _ProgressDots = require("./ProgressDots");

var _default = {
  title: 'ProgressDots',
  component: _ProgressDots.ProgressDots,
  argTypes: {
    steps: {
      control: {
        type: 'range',
        max: 10
      }
    }
  }
};
exports["default"] = _default;

var Basic = function Basic(args) {
  return /*#__PURE__*/_react["default"].createElement(_ProgressDots.ProgressDots, args);
};

exports.Basic = Basic;
var Loading = Basic.bind();
exports.Loading = Loading;
Loading.args = {
  isLoading: true
};
var Starting = Basic.bind();
exports.Starting = Starting;
Starting.args = {
  steps: 4,
  progress: 1
};
var Halfway = Basic.bind();
exports.Halfway = Halfway;
Halfway.args = {
  steps: 4,
  progress: 2
};
var Complete = Basic.bind();
exports.Complete = Complete;
Complete.args = {
  steps: 4,
  progress: 4
};
var LargeComplete = Basic.bind();
exports.LargeComplete = LargeComplete;
LargeComplete.args = {
  steps: 4,
  progress: 4,
  size: 'large'
};
