// @storybook/design-system/dist/components/Radio.stories.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Checked = exports.Unchecked = exports.All = exports.Template = exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _addonActions = require("@storybook/addon-actions");

var _Radio = require("./Radio");

var onChange = (0, _addonActions.action)('change');
var _default = {
  title: 'forms/Radio',
  component: _Radio.Radio
};
exports["default"] = _default;

var Template = function Template(args) {
  return /*#__PURE__*/_react["default"].createElement(_Radio.Radio, args);
};

exports.Template = Template;
Template.args = {
  label: 'label',
  hideLabel: false
};
Template.story = {
  name: 'Playground'
};

var All = function All() {
  return /*#__PURE__*/_react["default"].createElement("form", null, /*#__PURE__*/_react["default"].createElement(_Radio.Radio, {
    id: "Mice",
    label: "Mice",
    value: "mice",
    checked: true,
    onChange: onChange
  }), /*#__PURE__*/_react["default"].createElement(_Radio.Radio, {
    id: "Dogs",
    label: "Dogs",
    value: "dogs",
    onChange: onChange
  }), /*#__PURE__*/_react["default"].createElement(_Radio.Radio, {
    id: "Cats",
    label: "Cats",
    onChange: onChange,
    error: "There's a snake in my boots"
  }), /*#__PURE__*/_react["default"].createElement(_Radio.Radio, {
    id: "Dogs",
    label: "Dogs",
    description: "15 canines",
    value: "dogs",
    onChange: onChange
  }), /*#__PURE__*/_react["default"].createElement(_Radio.Radio, {
    id: "Mice",
    label: "Mice",
    value: "mice",
    onChange: onChange,
    checked: true,
    appearance: "secondary"
  }), /*#__PURE__*/_react["default"].createElement(_Radio.Radio, {
    id: "Dogs",
    label: "Dogs",
    value: "dogs",
    onChange: onChange,
    appearance: "secondary"
  }));
};

exports.All = All;
var Unchecked = Template.bind();
exports.Unchecked = Unchecked;
Unchecked.args = {
  id: 'Mice',
  label: 'Mice',
  hideLabel: true,
  value: 'mice'
};
var Checked = Template.bind();
exports.Checked = Checked;
Checked.args = {
  id: 'Dogs',
  label: 'Dogs',
  hideLabel: true,
  value: 'dogs',
  checked: true
};
