// @storybook/design-system/dist/components/Select.stories.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Tertiary = exports.Default = exports.All = exports.Template = exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _addonActions = require("@storybook/addon-actions");

var _Select = require("./Select");

var onChange = (0, _addonActions.action)('change');
var _default = {
  title: 'forms/Select',
  component: _Select.Select
};
exports["default"] = _default;

var Template = function Template(args) {
  return /*#__PURE__*/_react["default"].createElement(_Select.Select, args);
};

exports.Template = Template;
Template.args = {
  value: 'value1',
  options: [{
    label: 'Default',
    value: 'value1'
  }, {
    label: 'Dog',
    value: 'value2'
  }, {
    label: 'Mouse',
    value: 'value3'
  }]
};
Template.story = {
  name: 'Playground'
};

var All = function All() {
  return /*#__PURE__*/_react["default"].createElement("form", null, /*#__PURE__*/_react["default"].createElement(_Select.Select, {
    id: "Primary",
    value: "value1",
    options: [{
      label: 'Default',
      value: 'value1'
    }, {
      label: 'Dog',
      value: 'value2'
    }, {
      label: 'Mouse',
      value: 'value3'
    }],
    onChange: onChange,
    label: "Animal",
    hideLabel: true
  }), /*#__PURE__*/_react["default"].createElement(_Select.Select, {
    id: "Tertiary",
    value: "value1",
    options: [{
      label: 'Tertiary',
      value: 'value1'
    }, {
      label: 'Dog',
      value: 'value2'
    }, {
      label: 'Mouse',
      value: 'value3'
    }],
    appearance: "tertiary",
    onChange: onChange,
    label: "Animal",
    hideLabel: true
  }));
};

exports.All = All;

var Default = function Default() {
  return /*#__PURE__*/_react["default"].createElement("form", null, /*#__PURE__*/_react["default"].createElement(_Select.Select, {
    id: "Primary",
    label: "Animal",
    hideLabel: true,
    value: "value1",
    options: [{
      label: 'Default',
      value: 'value1'
    }, {
      label: 'Dog',
      value: 'value2'
    }],
    onChange: onChange
  }), /*#__PURE__*/_react["default"].createElement(_Select.Select, {
    id: "Primary-disabled",
    label: "Animal",
    hideLabel: true,
    value: "value1",
    options: [{
      label: 'Default',
      value: 'value1'
    }, {
      label: 'Dog',
      value: 'value2'
    }],
    disabled: true,
    onChange: onChange
  }), /*#__PURE__*/_react["default"].createElement(_Select.Select, {
    id: "Primary-with-iconn",
    label: "Animal",
    hideLabel: true,
    value: "value1",
    options: [{
      label: 'Default',
      value: 'value1'
    }, {
      label: 'Dog',
      value: 'value2'
    }],
    icon: "chromatic",
    onChange: onChange
  }), /*#__PURE__*/_react["default"].createElement(_Select.Select, {
    id: "Primary-in-progress",
    label: "Animal",
    hideLabel: true,
    value: "value1",
    options: [{
      label: 'Default',
      value: 'value1'
    }, {
      label: 'Dog',
      value: 'value2'
    }],
    icon: "chromatic",
    onChange: onChange,
    inProgress: true
  }), /*#__PURE__*/_react["default"].createElement(_Select.Select, {
    id: "Primary-with-error",
    label: "Animal",
    hideLabel: true,
    value: "value1",
    options: [{
      label: 'Default',
      value: 'value1'
    }, {
      label: 'Dog',
      value: 'value2'
    }],
    error: "There's a snake in my boots",
    onChange: onChange
  }), /*#__PURE__*/_react["default"].createElement(_Select.Select, {
    id: "Primary-with-icon-and-error",
    label: "Animal",
    hideLabel: true,
    value: "value1",
    options: [{
      label: 'Default',
      value: 'value1'
    }, {
      label: 'Dog',
      value: 'value2'
    }],
    icon: "chromatic",
    error: "There's a snake in my boots",
    onChange: onChange
  }), /*#__PURE__*/_react["default"].createElement(_Select.Select, {
    id: "Primary-with-label",
    value: "value1",
    options: [{
      label: 'Default',
      value: 'value1'
    }, {
      label: 'Dog',
      value: 'value2'
    }],
    icon: "chromatic",
    label: "Label",
    onChange: onChange
  }));
};

exports.Default = Default;

var Tertiary = function Tertiary() {
  return /*#__PURE__*/_react["default"].createElement("form", null, /*#__PURE__*/_react["default"].createElement(_Select.Select, {
    id: "Tertiary",
    label: "Animal",
    hideLabel: true,
    value: "value1",
    options: [{
      label: 'Default',
      value: 'value1'
    }, {
      label: 'Dog',
      value: 'value2'
    }],
    appearance: "tertiary",
    onChange: onChange
  }), /*#__PURE__*/_react["default"].createElement(_Select.Select, {
    id: "Tertiary-disabled",
    label: "Animal",
    hideLabel: true,
    value: "value1",
    options: [{
      label: 'Default',
      value: 'value1'
    }, {
      label: 'Dog',
      value: 'value2'
    }],
    disabled: true,
    appearance: "tertiary",
    onChange: onChange
  }));
};

exports.Tertiary = Tertiary;
