// @storybook/design-system/dist/components/SharedStyles.stories.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.HoverEffectActive = exports.HoverEffectHover = exports.HoverEffectRest = exports.PageMargins = exports["default"] = void 0;

var _taggedTemplateLiteral2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteral"));

var _react = _interopRequireDefault(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _styles = require("./shared/styles");

function _templateObject2() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  ", ";\n  display: block;\n  height: 300px;\n"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  ", ";\n  height: 300px;\n  background-color: #333;\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

var _default = {
  title: 'SharedStyles'
};
exports["default"] = _default;

var BlockWithMargin = _styledComponents["default"].div(_templateObject(), _styles.pageMargins);

var PageMargins = function PageMargins() {
  return /*#__PURE__*/_react["default"].createElement("div", null, /*#__PURE__*/_react["default"].createElement("p", null, "The box below has ", /*#__PURE__*/_react["default"].createElement("code", null, "pageMargins"), " styles applied to it which controls the horizontal padding and margin"), /*#__PURE__*/_react["default"].createElement(BlockWithMargin, null));
};

exports.PageMargins = PageMargins;

var BlockWithHoverEffect = _styledComponents["default"].div(_templateObject2(), _styles.hoverEffect);

var HoverEffectRest = function HoverEffectRest() {
  return /*#__PURE__*/_react["default"].createElement(BlockWithHoverEffect, null);
};

exports.HoverEffectRest = HoverEffectRest;

var HoverEffectHover = function HoverEffectHover() {
  return /*#__PURE__*/_react["default"].createElement(BlockWithHoverEffect, {
    className: "__hover"
  });
};

exports.HoverEffectHover = HoverEffectHover;

var HoverEffectActive = function HoverEffectActive() {
  return /*#__PURE__*/_react["default"].createElement(BlockWithHoverEffect, {
    className: "__active"
  });
};

exports.HoverEffectActive = HoverEffectActive;
