// @storybook/design-system/dist/components/Spinner.stories.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Inline = exports.InForm = exports.Inverse = exports.Default = exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _Spinner = require("./Spinner");

var _default = {
  title: 'Spinner',
  component: _Spinner.Spinner
};
exports["default"] = _default;

var Default = function Default() {
  return /*#__PURE__*/_react["default"].createElement("div", {
    style: {
      background: 'white',
      padding: '1.5em',
      position: 'relative'
    }
  }, /*#__PURE__*/_react["default"].createElement(_Spinner.Spinner, null));
};

exports.Default = Default;

var Inverse = function Inverse() {
  return /*#__PURE__*/_react["default"].createElement("div", {
    style: {
      background: 'black',
      padding: '1.5em',
      position: 'relative'
    }
  }, /*#__PURE__*/_react["default"].createElement(_Spinner.Spinner, {
    inverse: true
  }));
};

exports.Inverse = Inverse;

var InForm = function InForm() {
  return /*#__PURE__*/_react["default"].createElement("div", {
    style: {
      background: '#eee',
      padding: '1.5em',
      position: 'relative'
    }
  }, /*#__PURE__*/_react["default"].createElement(_Spinner.Spinner, {
    inForm: true
  }));
};

exports.InForm = InForm;

var Inline = function Inline() {
  return /*#__PURE__*/_react["default"].createElement("div", {
    style: {
      background: '#eee',
      padding: '1.5em',
      position: 'relative'
    }
  }, /*#__PURE__*/_react["default"].createElement(_Spinner.Spinner, {
    inline: true
  }));
};

exports.Inline = Inline;
