// @storybook/design-system/dist/components/Textarea.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Textarea = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _taggedTemplateLiteral2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteral"));

var _react = _interopRequireWildcard(require("react"));

var _propTypes = _interopRequireDefault(require("prop-types"));

var _styledComponents = _interopRequireWildcard(require("styled-components"));

var _styles = require("./shared/styles");

function _templateObject12() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n      display: table-row;\n\n      ", ", ", " {\n        display: table-cell;\n      }\n\n      ", " {\n        width: 1px;\n        padding-right: 20px;\n      }\n\n      ", " {\n        width: auto;\n      }\n    "]);

  _templateObject12 = function _templateObject12() {
    return data;
  };

  return data;
}

function _templateObject11() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  ", "\n"]);

  _templateObject11 = function _templateObject11() {
    return data;
  };

  return data;
}

function _templateObject10() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n      ", " {\n        box-shadow: ", " 0 0 0 1px inset;\n        &:focus {\n          box-shadow: ", " 0 0 0 1px inset !important;\n        }\n      }\n    "]);

  _templateObject10 = function _templateObject10() {
    return data;
  };

  return data;
}

function _templateObject9() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  display: inline-block;\n  font-size: ", "px;\n  overflow: hidden;\n  position: relative;\n  vertical-align: top;\n  width: 100%;\n\n  ", " {\n    background: ", ";\n    border-radius: 0;\n    color: ", ";\n    font-size: ", "px;\n    line-height: 20px;\n    padding: 0.7111em 1em; //40\n\n    box-shadow: ", " 0 0 0 1px inset;\n    &:focus {\n      box-shadow: ", " 0 0 0 1px inset;\n    }\n  }\n\n  ", ";\n"]);

  _templateObject9 = function _templateObject9() {
    return data;
  };

  return data;
}

function _templateObject8() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  ::placeholder {\n    color: ", ";\n  }\n  appearance: none;\n  border: none;\n  box-sizing: border-box;\n  display: block;\n  outline: none;\n  width: 100%;\n  word-wrap: break-word;\n\n  &[disabled] {\n    cursor: not-allowed;\n    opacity: 0.5;\n  }\n\n  &:-webkit-autofill {\n    -webkit-box-shadow: 0 0 0 3em ", " inset;\n  }\n"]);

  _templateObject8 = function _templateObject8() {
    return data;
  };

  return data;
}

function _templateObject7() {
  var data = (0, _taggedTemplateLiteral2["default"])([""]);

  _templateObject7 = function _templateObject7() {
    return data;
  };

  return data;
}

function _templateObject6() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n      ", " {\n        margin-left: 0.5em;\n      }\n    "]);

  _templateObject6 = function _templateObject6() {
    return data;
  };

  return data;
}

function _templateObject5() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n      height: 0;\n      margin: 0;\n    "]);

  _templateObject5 = function _templateObject5() {
    return data;
  };

  return data;
}

function _templateObject4() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  margin-bottom: 0.33em;\n  font-weight: ", ";\n  font-size: ", "px;\n\n  ", "\n\n  ", "\n"]);

  _templateObject4 = function _templateObject4() {
    return data;
  };

  return data;
}

function _templateObject3() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  color: ", ";\n  font-weight: ", ";\n"]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n      border: 0px !important;\n      clip: rect(0 0 0 0) !important;\n      -webkit-clip-path: inset(100%) !important;\n      clip-path: inset(100%) !important;\n      height: 1px !important;\n      overflow: hidden !important;\n      padding: 0px !important;\n      position: absolute !important;\n      white-space: nowrap !important;\n      width: 1px !important;\n    "]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = (0, _taggedTemplateLiteral2["default"])(["\n  ", "\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

var Label = _styledComponents["default"].label(_templateObject(), function (props) {
  return props.hideLabel && (0, _styledComponents.css)(_templateObject2());
});

var ErrorMessage = _styledComponents["default"].span(_templateObject3(), _styles.color.negative, _styles.typography.weight.regular);

var LabelWrapper = _styledComponents["default"].div(_templateObject4(), _styles.typography.weight.bold, _styles.typography.size.s2, function (props) {
  return props.hideLabel && !props.error && (0, _styledComponents.css)(_templateObject5());
}, function (props) {
  return !props.hideLabel && (0, _styledComponents.css)(_templateObject6(), ErrorMessage);
});

var Subtext = _styledComponents["default"].div(_templateObject7());

var TextareaText = _styledComponents["default"].textarea(_templateObject8(), _styles.color.mediumdark, _styles.color.lightest);

var TextareaWrapper = _styledComponents["default"].div(_templateObject9(), _styles.typography.size.s2, TextareaText, _styles.color.lightest, _styles.color.darkest, _styles.typography.size.s2, _styles.color.border, _styles.color.secondary, function (props) {
  return props.error && (0, _styledComponents.css)(_templateObject10(), TextareaText, _styles.color.negative, _styles.color.negative);
});

var TextareaContainer = _styledComponents["default"].div(_templateObject11(), function (props) {
  return props.orientation === 'horizontal' && (0, _styledComponents.css)(_templateObject12(), LabelWrapper, TextareaWrapper, LabelWrapper, TextareaWrapper);
});

var getErrorMessage = function getErrorMessage(_ref) {
  var error = _ref.error,
      value = _ref.value;
  return typeof error === 'function' ? error(value) : error;
};

var Textarea = /*#__PURE__*/(0, _react.forwardRef)(function (_ref2, ref) {
  var id = _ref2.id,
      value = _ref2.value,
      label = _ref2.label,
      hideLabel = _ref2.hideLabel,
      error = _ref2.error,
      subtext = _ref2.subtext,
      subtextSentiment = _ref2.subtextSentiment,
      appearance = _ref2.appearance,
      orientation = _ref2.orientation,
      className = _ref2.className,
      startFocused = _ref2.startFocused,
      rest = (0, _objectWithoutProperties2["default"])(_ref2, ["id", "value", "label", "hideLabel", "error", "subtext", "subtextSentiment", "appearance", "orientation", "className", "startFocused"]);
  // Outside refs take precedence
  var textareaRef = ref || (0, _react.useRef)();
  var didFocusOnStart = (0, _react.useRef)(false);
  (0, _react.useEffect)(function () {
    if (textareaRef && textareaRef.current && startFocused && !didFocusOnStart.current) {
      textareaRef.current.focus();
      didFocusOnStart.current = true;
    }
  }, [textareaRef, textareaRef.current, didFocusOnStart, didFocusOnStart.current]);

  var _useState = (0, _react.useState)(getErrorMessage({
    error: error,
    value: value
  })),
      _useState2 = (0, _slicedToArray2["default"])(_useState, 2),
      errorMessage = _useState2[0],
      setErrorMessage = _useState2[1];

  (0, _react.useEffect)(function () {
    setErrorMessage(getErrorMessage({
      error: error,
      value: value
    }));
  }, [value, error]);
  var errorId = "".concat(id, "-error");
  var subtextId = "".concat(id, "-subtext");
  var ariaDescribedBy = "".concat(error ? errorId : '', " ").concat(subtext ? subtextId : '');
  return /*#__PURE__*/_react["default"].createElement(TextareaContainer, {
    orientation: orientation,
    className: className
  }, /*#__PURE__*/_react["default"].createElement(LabelWrapper, {
    appearance: appearance,
    hideLabel: hideLabel,
    error: errorMessage
  }, /*#__PURE__*/_react["default"].createElement(Label, {
    htmlFor: id,
    hideLabel: hideLabel
  }, label), errorMessage && /*#__PURE__*/_react["default"].createElement(ErrorMessage, {
    id: errorId,
    "aria-hidden": true
  }, errorMessage)), /*#__PURE__*/_react["default"].createElement(TextareaWrapper, {
    error: errorMessage,
    appearance: appearance
  }, /*#__PURE__*/_react["default"].createElement(TextareaText, (0, _extends2["default"])({
    id: id,
    value: value,
    rows: "7",
    "aria-invalid": !!error,
    "aria-describedby": ariaDescribedBy,
    ref: textareaRef
  }, rest)), subtext && /*#__PURE__*/_react["default"].createElement(Subtext, {
    id: subtextId,
    sentiment: subtextSentiment
  }, subtext)));
});
exports.Textarea = Textarea;
Textarea.propTypes = {
  id: _propTypes["default"].string.isRequired,
  value: _propTypes["default"].string.isRequired,
  appearance: _propTypes["default"].oneOf(['default']),
  label: _propTypes["default"].string.isRequired,
  hideLabel: _propTypes["default"].bool,
  orientation: _propTypes["default"].oneOf(['vertical', 'horizontal']),
  error: _propTypes["default"].oneOfType([_propTypes["default"].node, _propTypes["default"].func]),
  startFocused: _propTypes["default"].bool,
  subtext: _propTypes["default"].string,
  subtextSentiment: _propTypes["default"].oneOf(['default', 'negative', 'warning']),
  className: _propTypes["default"].string
};
Textarea.defaultProps = {
  appearance: 'default',
  hideLabel: false,
  orientation: 'vertical',
  error: null,
  startFocused: false,
  subtext: null,
  subtextSentiment: 'default',
  className: null
};
