// @storybook/design-system/dist/components/modal/WithModal.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = WithModal;

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var _react = _interopRequireWildcard(require("react"));

var _propTypes = require("prop-types");

var _Modal = require("./Modal");

function PureWithModal(_ref) {
  var isOpen = _ref.isOpen,
      onOpen = _ref.onOpen,
      onClose = _ref.onClose,
      modal = _ref.modal,
      children = _ref.children,
      rest = (0, _objectWithoutProperties2["default"])(_ref, ["isOpen", "onOpen", "onClose", "modal", "children"]);
  return /*#__PURE__*/_react["default"].createElement(_react["default"].Fragment, null, _react.Children.only(children({
    isOpen: isOpen,
    onOpen: onOpen,
    onClose: onClose
  })), /*#__PURE__*/_react["default"].createElement(_Modal.Modal, (0, _extends2["default"])({
    isOpen: isOpen,
    onClose: onClose
  }, rest), modal));
}

PureWithModal.propTypes = {
  isOpen: _propTypes.PropTypes.bool.isRequired,
  onOpen: _propTypes.PropTypes.func.isRequired,
  onClose: _propTypes.PropTypes.func.isRequired,
  modal: _propTypes.PropTypes.func.isRequired,
  children: _propTypes.PropTypes.func.isRequired
};

function WithModal(_ref2) {
  var startOpen = _ref2.startOpen,
      onOpen = _ref2.onOpen,
      onClose = _ref2.onClose,
      rest = (0, _objectWithoutProperties2["default"])(_ref2, ["startOpen", "onOpen", "onClose"]);

  var _useState = (0, _react.useState)(startOpen),
      _useState2 = (0, _slicedToArray2["default"])(_useState, 2),
      isOpen = _useState2[0],
      onSetIsOpen = _useState2[1];

  var handleOpen = function handleOpen() {
    onSetIsOpen(true);

    if (onOpen) {
      onOpen();
    }
  };

  var handleClose = function handleClose() {
    onSetIsOpen(false);

    if (onClose) {
      onClose();
    }
  };

  return /*#__PURE__*/_react["default"].createElement(PureWithModal, (0, _extends2["default"])({
    isOpen: isOpen,
    onOpen: handleOpen,
    onClose: handleClose
  }, rest));
}

WithModal.propTypes = {
  startOpen: _propTypes.PropTypes.bool,
  onClose: _propTypes.PropTypes.func,
  onOpen: _propTypes.PropTypes.func
};
WithModal.defaultProps = {
  startOpen: false,
  onClose: undefined,
  onOpen: undefined
};
